<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>基于GCTA_fastBAT的gene-based关联分析研究 — GCTA_fastBAT • QTLMR</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="基于GCTA_fastBAT的gene-based关联分析研究 — GCTA_fastBAT"><meta name="description" content="该函数用于执行GCTA软件包中的FastBAT分析，基于给定的GWAS数据文件和基因列表文件，进行快速的基因-表型关联分析，详见：https://yanglab.westlake.edu.cn/software/gcta/#fastBAT。"><meta property="og:description" content="该函数用于执行GCTA软件包中的FastBAT分析，基于给定的GWAS数据文件和基因列表文件，进行快速的基因-表型关联分析，详见：https://yanglab.westlake.edu.cn/software/gcta/#fastBAT。"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">QTLMR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">2.43</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../index.html">主页</a></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">函数功能</a></li>
<li class="nav-item"><a class="nav-link" href="../articles/index.html">文章示例代码</a></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">更新日志</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>基于GCTA_fastBAT的gene-based关联分析研究</h1>

      <div class="d-none name"><code>GCTA_fastBAT.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>该函数用于执行GCTA软件包中的FastBAT分析，基于给定的GWAS数据文件和基因列表文件，进行快速的基因-表型关联分析，详见：https://yanglab.westlake.edu.cn/software/gcta/#fastBAT。</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">GCTA_fastBAT</span><span class="op">(</span></span>
<span>  GWSAfile <span class="op">=</span> <span class="st">"./Migraine_SMR_FinnGen.ma"</span>,</span>
<span>  bfile <span class="op">=</span> <span class="st">"./1.data/1000G/EUR"</span>,</span>
<span>  gene_list <span class="op">=</span> <span class="st">"./glist-hg38"</span>,</span>
<span>  maf <span class="op">=</span> <span class="fl">0.01</span>,</span>
<span>  cores <span class="op">=</span> <span class="fl">6</span>,</span>
<span>  opt_arguments <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  save_name <span class="op">=</span> <span class="st">"Migraine"</span>,</span>
<span>  save_path <span class="op">=</span> <span class="st">"fastBAT"</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-gwsafile">GWSAfile<a class="anchor" aria-label="anchor" href="#arg-gwsafile"></a></dt>
<dd><p>字符串，指定输入.ma格式的GWAS结果文件路径（如：<code>Migraine_SMR_FinnGen.ma</code>）。</p></dd>


<dt id="arg-bfile">bfile<a class="anchor" aria-label="anchor" href="#arg-bfile"></a></dt>
<dd><p>字符串，进行基因型数据的连锁不平衡（LD）估计时的参考面板文件的前缀路径（如：<code>./1.data/1000G/EUR</code>）。</p></dd>


<dt id="arg-gene-list">gene_list<a class="anchor" aria-label="anchor" href="#arg-gene-list"></a></dt>
<dd><p>字符串，指定基因列表文件的路径（如：<code>./glist-hg38</code>）。</p></dd>


<dt id="arg-maf">maf<a class="anchor" aria-label="anchor" href="#arg-maf"></a></dt>
<dd><p>数值，指定过滤的最小等位基因频率（MAF），默认值为0.01。</p></dd>


<dt id="arg-cores">cores<a class="anchor" aria-label="anchor" href="#arg-cores"></a></dt>
<dd><p>数值，指定要使用的CPU线程数量，默认值为2。</p></dd>


<dt id="arg-opt-arguments">opt_arguments<a class="anchor" aria-label="anchor" href="#arg-opt-arguments"></a></dt>
<dd><p>字符串或NULL，额外的命令行参数，用于GCTA的FastBAT分析（默认为NULL）。</p></dd>


<dt id="arg-save-name">save_name<a class="anchor" aria-label="anchor" href="#arg-save-name"></a></dt>
<dd><p>字符串，指定输出结果文件的名称。</p></dd>


<dt id="arg-save-path">save_path<a class="anchor" aria-label="anchor" href="#arg-save-path"></a></dt>
<dd><p>字符串，指定结果文件保存路径。</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>返回.gene.fbat结果，结果文件包含以下内容：</p>
<dl><dt>Gene</dt>
<dd><p>基因ID</p></dd>

<dt>Chr</dt>
<dd><p>染色体编号</p></dd>

<dt>Start and End</dt>
<dd><p>基因区域的左侧和右侧边界</p></dd>

<dt>No.SNPs</dt>
<dd><p>基因区域内的SNP数量</p></dd>

<dt>SNP_start and SNP_end</dt>
<dd><p>基因区域的左侧和右侧边界SNP位置</p></dd>

<dt>Chisq(Obs)</dt>
<dd><p>基因区域内所有SNP的卡方检验统计量之和</p></dd>

<dt>Pvalue</dt>
<dd><p>基因水平的检验p值</p></dd>

<dt>TopSNP.Pvalue</dt>
<dd><p>基因区域内最小的单个SNP GWAS p值</p></dd>

<dt>TopSNP</dt>
<dd><p>与基因关联的最显著GWAS SNP</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="note">Note<a class="anchor" aria-label="anchor" href="#note"></a></h2>
    <p>该方法利用全基因组关联研究（GWAS）的汇总数据和来自参考样本的个体级基因型数据的连锁不平衡（LD）数据，
执行人类复杂性状的快速集合关联分析。有关该方法的详细信息，请参阅Bakshi等人（2016年，Scientific Reports）的研究，PMID: 27604177。</p>
    </div>

  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by YB H.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

    </footer></div>





  </body></html>

