<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Package index • QTLMR</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Package index"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">QTLMR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">2.34</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../index.html">主页</a></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">函数功能</a></li>
<li class="nav-item"><a class="nav-link" href="../articles/index.html">文章示例代码</a></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">更新日志</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-index">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Package index</h1>
    </div>

    <div class="section level2">

      <h3 id="-r-">1.R包配置<a class="anchor" aria-label="anchor" href="#-r-"></a></h3>



    </div><div class="section level2">




      <dl><dt>

          <code><a href="install_Rtools.html">install_Rtools()</a></code>

        </dt>
        <dd>安装Rtools软件</dd>
      </dl></div><div class="section level2">

      <h3 id="-">2.百度网盘下载文件<a class="anchor" aria-label="anchor" href="#-"></a></h3>



    </div><div class="section level2">




      <dl><dt>

          <code><a href="Baidu_py_install.html">Baidu_py_install()</a></code>

        </dt>
        <dd>配置百度网盘python环境</dd>
      </dl><dl><dt>

          <code><a href="Baidu_download.html">Baidu_download()</a></code>

        </dt>
        <dd>使用百度网盘下载文件</dd>
      </dl></div><div class="section level2">

      <h3 id="--1">3.内置组学数据<a class="anchor" aria-label="anchor" href="#--1"></a></h3>



    </div><div class="section level2">




      <dl><dt>

          <code><a href="available_outcomes_ieu.html">available_outcomes_ieu()</a></code>

        </dt>
        <dd>IEU Open GWAS数据库表型汇总表</dd>
      </dl><dl><dt>

          <code><a href="data_info_ukb_ppp_pQTL.html">data_info_ukb_ppp_pQTL()</a></code>

        </dt>
        <dd>ukb-ppp pQTL汇总信息数据</dd>
      </dl><dl><dt>

          <code><a href="data_cispQTL_ukb_ppp_exp.html">data_cispQTL_ukb_ppp_exp()</a></code>

        </dt>
        <dd>ukb-ppp cis-pQTL的暴露数据</dd>
      </dl><dl><dt>

          <code><a href="data_info_decode_pQTL.html">data_info_decode_pQTL()</a></code>

        </dt>
        <dd>decode pQTL汇总信息数据</dd>
      </dl><dl><dt>

          <code><a href="data_cispQT_decode_exp.html">data_cispQT_decode_exp()</a></code>

        </dt>
        <dd>decode cis-pQTL的暴露数据</dd>
      </dl><dl><dt>

          <code><a href="data_info_fenland_pQTL.html">data_info_fenland_pQTL()</a></code>

        </dt>
        <dd>fenland pQTL汇总信息数据</dd>
      </dl><dl><dt>

          <code><a href="data_cispQT_fenland_exp.html">data_cispQT_fenland_exp()</a></code>

        </dt>
        <dd>fenland cis-pQTL的暴露数据</dd>
      </dl><dl><dt>

          <code><a href="data_Gut_microbes_196.html">data_Gut_microbes_196()</a></code>

        </dt>
        <dd>196种肠道微生物暴露数据</dd>
      </dl><dl><dt>

          <code><a href="data_Gut_microbes_412.html">data_Gut_microbes_412()</a></code>

        </dt>
        <dd>412种肠道微生物暴露数据</dd>
      </dl><dl><dt>

          <code><a href="data_Gut_microbes_473.html">data_Gut_microbes_473()</a></code>

        </dt>
        <dd>473种肠道微生物暴露数据</dd>
      </dl><dl><dt>

          <code><a href="data_Immunocytokine_731.html">data_Immunocytokine_731()</a></code>

        </dt>
        <dd>731种免疫细胞因子暴露数据</dd>
      </dl><dl><dt>

          <code><a href="data_Inflammatory_factors_41.html">data_Inflammatory_factors_41()</a></code>

        </dt>
        <dd>41种炎症因子暴露数据</dd>
      </dl><dl><dt>

          <code><a href="data_Inflammatory_proteins_91.html">data_Inflammatory_proteins_91()</a></code>

        </dt>
        <dd>91种血浆循环炎症蛋白</dd>
      </dl><dl><dt>

          <code><a href="data_IEU_prot_a.html">data_IEU_prot_a()</a></code>

        </dt>
        <dd>IEU数据库中Sun_BB_protein的3282个蛋白数据</dd>
      </dl><dl><dt>

          <code><a href="data_IEU_prot_c.html">data_IEU_prot_c()</a></code>

        </dt>
        <dd>IEU数据库中prot_c的1124个蛋白数据</dd>
      </dl><dl><dt>

          <code><a href="data_Metabolites_1400.html">data_Metabolites_1400()</a></code>

        </dt>
        <dd>1400种代谢产物的暴露数据</dd>
      </dl><dl><dt>

          <code><a href="data_IEU_exposure_dat.html">data_IEU_exposure_dat()</a></code>

        </dt>
        <dd>IEU数据库中7020个临床表型暴露数据</dd>
      </dl><dl><dt>

          <code><a href="data_info_finngen.html">data_info_finngen()</a></code>

        </dt>
        <dd>芬兰R7-R12汇总信息数据</dd>
      </dl><dl><dt>

          <code><a href="data_info_ukb_neale.html">data_info_ukb_neale()</a></code>

        </dt>
        <dd>ukb neale汇总信息数据</dd>
      </dl><dl><dt>

          <code><a href="data_info_GTEx_V8_samplesize.html">data_info_GTEx_V8_samplesize()</a></code>

        </dt>
        <dd>GTEx_V8样本量汇总信息数据</dd>
      </dl><dl><dt>

          <code><a href="data_info_GTEx_V8_GENE.html">data_info_GTEx_V8_GENE()</a></code>

        </dt>
        <dd>GTEx_V8基因信息汇总数据</dd>
      </dl><dl><dt>

          <code><a href="data_info_cytoBand.html">data_info_cytoBand()</a></code>

        </dt>
        <dd>cytoBand汇总信息数据</dd>
      </dl></div><div class="section level2">

      <h3 id="-xqtl-">4.在线获取xQTL或药靶数据<a class="anchor" aria-label="anchor" href="#-xqtl-"></a></h3>



    </div><div class="section level2">




      <dl><dt>

          <code><a href="get_cispQTL_decode_ukb_Online.html">get_cispQTL_decode_ukb_Online()</a></code>

        </dt>
        <dd>在线获取"decode"或"ukb_ppp"的顺式pQTL数据</dd>
      </dl><dl><dt>

          <code><a href="get_cispQTL_fenland_Online.html">get_cispQTL_fenland_Online()</a></code>

        </dt>
        <dd>在线获取fenland的顺式pQTL数据</dd>
      </dl><dl><dt>

          <code><a href="get_ciseQTL_eQTLGen_Online.html">get_ciseQTL_eQTLGen_Online()</a></code>

        </dt>
        <dd>在线从IEU数据库获取的eQTLGen顺式eQTL数据</dd>
      </dl><dl><dt>

          <code><a href="get_ciseQTL_GTEx_V8_Online.html">get_ciseQTL_GTEx_V8_Online()</a></code>

        </dt>
        <dd>在线获取GTEx_V8顺式eqtl数据</dd>
      </dl><dl><dt>

          <code><a href="get_Drugs_gene_IVs.html">get_Drugs_gene_IVs()</a></code>

        </dt>
        <dd>获取传统药靶基因工具变量</dd>
      </dl></div><div class="section level2">

      <h3 id="--2">5.获取基因信息<a class="anchor" aria-label="anchor" href="#--2"></a></h3>



    </div><div class="section level2">




      <dl><dt>

          <code><a href="Gene_start_end_info_ensembl.html">Gene_start_end_info_ensembl()</a></code>

        </dt>
        <dd>从ensembl数据库中获取基因的信息</dd>
      </dl><dl><dt>

          <code><a href="Gene_start_end_info_GTF.html">Gene_start_end_info_GTF()</a></code>

        </dt>
        <dd>从gencodegenes的gtf.gz原始数据获取基因信息</dd>
      </dl><dl><dt>

          <code><a href="Gene_start_end_info_modified.html">Gene_start_end_info_modified()</a></code>

        </dt>
        <dd>从内置数据中获取Ensembl id与Gene汇总信息数据</dd>
      </dl></div><div class="section level2">

      <h3 id="--3">6.数据转换<a class="anchor" aria-label="anchor" href="#--3"></a></h3>



    </div><div class="section level2">




      <dl><dt>

          <code><a href="format_dat.html">format_dat()</a></code>

        </dt>
        <dd>数据格式转换与清洗</dd>
      </dl><dl><dt>

          <code><a href="format_data_FinnGen.html">format_data_FinnGen()</a></code>

        </dt>
        <dd>将FinnGen数据库.gz文件转换为标准TwosampleMR、SMR、METAL、MTAG分析文件</dd>
      </dl><dl><dt>

          <code><a href="format_data_Catalog.html">format_data_Catalog()</a></code>

        </dt>
        <dd>将GWAS Catalog数据库.gz或.tsv文件转换为标准TwosampleMR、SMR、METAL、MTAG分析文件</dd>
      </dl><dl><dt>

          <code><a href="format_data_ukb_Neale.html">format_data_ukb_Neale()</a></code>

        </dt>
        <dd>将ukb nealelab数据库tsv.bgz文件转换为标准TwosampleMR、SMR、METAL、MTAG分析文件</dd>
      </dl><dl><dt>

          <code><a href="format_data_IEU_VCF.html">format_data_IEU_VCF()</a></code>

        </dt>
        <dd>在线或本地读取IEU数据库.VCF文件转换为标准TwosampleMR、SMR、METAL、MTAG分析文件</dd>
      </dl><dl><dt>

          <code><a href="format_data_METAL.html">format_data_METAL()</a></code>

        </dt>
        <dd>将METAL分析结果数据转换为标准TwosampleMR、SMR文件</dd>
      </dl><dl><dt>

          <code><a href="format_data_decode.html">format_data_decode()</a></code>

        </dt>
        <dd>将decode原始数据txt.gz文件重注释转换为标准TwosampleMR、SMR、coloc共定位文件</dd>
      </dl><dl><dt>

          <code><a href="format_data_ukb_pQTL_cis.html">format_data_ukb_pQTL_cis()</a></code>

        </dt>
        <dd>将ukb_ppp基因所在染色体的数据.gz文件重注释转换为标准SMR、coloc共定位文件</dd>
      </dl><dl><dt>

          <code><a href="format_data_fenland.html">format_data_fenland()</a></code>

        </dt>
        <dd>将fenland pQTL原始数据txt.gz文件为标准TwosampleMR、SMR、coloc共定位文件</dd>
      </dl><dl><dt>

          <code><a href="GWAS_liftover.html">GWAS_liftover()</a></code>

        </dt>
        <dd>使用liftover转换GWAS数据的基因组版本</dd>
      </dl><dl><dt>

          <code><a href="xQTL_cis_dat.html">xQTL_cis_dat()</a></code>

        </dt>
        <dd>提取原始本地xQTL数据的顺式区域数据</dd>
      </dl></div><div class="section level2">

      <h3 id="-snp-chrpos-">7.SNP与chrpos互相转换<a class="anchor" aria-label="anchor" href="#-snp-chrpos-"></a></h3>



    </div><div class="section level2">




      <dl><dt>

          <code><a href="tran_chrpos_from_SNP.html">tran_chrpos_from_SNP()</a></code>

        </dt>
        <dd>rsid转chrpos</dd>
      </dl><dl><dt>

          <code><a href="tran_SNP_from_chrpos.html">tran_SNP_from_chrpos()</a></code>

        </dt>
        <dd>chrpos转rsid</dd>
      </dl><dl><dt>

          <code><a href="tran_SNP_python.html">tran_SNP_python()</a></code>

        </dt>
        <dd>使用python实现chrpos转rsid</dd>
      </dl></div><div class="section level2">

      <h3 id="-xqtl--1">8.xQTL相关孟德尔随机化分析<a class="anchor" aria-label="anchor" href="#-xqtl--1"></a></h3>



    </div><div class="section level2">




      <dl><dt>

          <code><a href="remove_MHC_data.html">remove_MHC_data()</a></code>

        </dt>
        <dd>移除MHC区域的SNP数据</dd>
      </dl><dl><dt>

          <code><a href="calculation_Fvalue_R2.html">calculation_Fvalue_R2()</a></code>

        </dt>
        <dd>计算F值与R2</dd>
      </dl><dl><dt>

          <code><a href="clump_data_local_Online.html">clump_data_local_Online()</a></code>

        </dt>
        <dd>本地或在线方式去除连锁不平衡</dd>
      </dl><dl><dt>

          <code><a href="modified_harmonise_proxy_Ensembl.html">modified_harmonise_proxy_Ensembl()</a></code>

        </dt>
        <dd>本地数据harmonise并基于Ensembl数据库代理SNP</dd>
      </dl><dl><dt>

          <code><a href="modified_proxy_1000G.html">modified_proxy_1000G()</a></code>

        </dt>
        <dd>本地数据提取结局数据并基于1000G代理SNP</dd>
      </dl><dl><dt>

          <code><a href="modified_Proxy_SNP_LDlink.html">modified_Proxy_SNP_LDlink()</a></code>

        </dt>
        <dd>本地数据提取结局数据并基于LDlink代理SNP</dd>
      </dl><dl><dt>

          <code><a href="xQTL_mr.html">xQTL_mr()</a></code>

        </dt>
        <dd>xQTL数据进行MR分析</dd>
      </dl><dl><dt>

          <code><a href="xQTL_volcano_plot.html">xQTL_volcano_plot()</a></code>

        </dt>
        <dd>绘制xQTL孟德尔随机化结果火山图</dd>
      </dl><dl><dt>

          <code><a href="xQTL_forest.html">xQTL_forest()</a></code>

        </dt>
        <dd>绘制xQTL森林图（Forest Plot）</dd>
      </dl></div><div class="section level2">

      <h3 id="--4">9.共定位分析<a class="anchor" aria-label="anchor" href="#--4"></a></h3>



    </div><div class="section level2">




      <dl><dt>

          <code><a href="coloc_format_dat.html">coloc_format_dat()</a></code>

        </dt>
        <dd>共定位基因组区域数据提取</dd>
      </dl><dl><dt>

          <code><a href="coloc_abf.html">coloc_abf()</a></code>

        </dt>
        <dd>coloc共定位分析并绘制locuscomparer图</dd>
      </dl><dl><dt>

          <code><a href="coloc_susie.html">coloc_susie()</a></code>

        </dt>
        <dd>SuSiE（单一效应之和）共定位分析</dd>
      </dl><dl><dt>

          <code><a href="PWCOCO_run.html">PWCOCO_run()</a></code>

        </dt>
        <dd>使用PWCOCO进行成对条件与共定位分析</dd>
      </dl><dl><dt>

          <code><a href="coloc_stack_assoc_plot.html">coloc_stack_assoc_plot()</a></code>

        </dt>
        <dd>绘制stack_assoc_plot图</dd>
      </dl></div><div class="section level2">

      <h3 id="-smr-">10.SMR分析<a class="anchor" aria-label="anchor" href="#-smr-"></a></h3>



    </div><div class="section level2">




      <dl><dt>

          <code><a href="SMR_format_BESD.html">SMR_format_BESD()</a></code>

        </dt>
        <dd>构建SMR分析所需的BESD二进制数据文件</dd>
      </dl><dl><dt>

          <code><a href="SMR_multi_HEIDI.html">SMR_multi_HEIDI()</a></code>

        </dt>
        <dd>基于SMR软件的孟德尔随机化分析</dd>
      </dl><dl><dt>

          <code><a href="SMR_Locus_Effect_Plot.html">SMR_Locus_Effect_Plot()</a></code>

        </dt>
        <dd>绘制SMR locus与Effect Plot图</dd>
      </dl></div><div class="section level2">

      <h3 id="-gsmr-">11.GSMR分析<a class="anchor" aria-label="anchor" href="#-gsmr-"></a></h3>



    </div><div class="section level2">




      <dl><dt>

          <code><a href="GSMR2_QTLMR.html">GSMR2_QTLMR()</a></code>

        </dt>
        <dd>使用GSMR基于广义汇总数据的孟德尔随机化分析</dd>
      </dl></div><div class="section level2">

      <h3 id="-ldsc-">12.LDSC分析<a class="anchor" aria-label="anchor" href="#-ldsc-"></a></h3>



    </div><div class="section level2">




      <dl><dt>

          <code><a href="LDSC_MRlap.html">LDSC_MRlap()</a></code>

        </dt>
        <dd>使用MRlap包分析交叉性状遗传相关性，矫正样本重叠引起的IVW-MR分析的偏差</dd>
      </dl><dl><dt>

          <code><a href="LDSC_ldscr_h2.html">LDSC_ldscr_h2()</a></code>

        </dt>
        <dd>使用ldscr包计算遗传力</dd>
      </dl><dl><dt>

          <code><a href="LDSC_ldscr_rg.html">LDSC_ldscr_rg()</a></code>

        </dt>
        <dd>使用ldscr包计算交叉性状遗传相关性</dd>
      </dl><dl><dt>

          <code><a href="LDSC_py_install.html">LDSC_py_install()</a></code>

        </dt>
        <dd>配置ldsc软件环境</dd>
      </dl><dl><dt>

          <code><a href="LDSC_py_sumstats.html">LDSC_py_sumstats()</a></code>

        </dt>
        <dd>使用ldsc软件生成.sumstats.gz格式文件(表头：SNP A1 A2 N Z)</dd>
      </dl><dl><dt>

          <code><a href="LDSC_py_h2.html">LDSC_py_h2()</a></code>

        </dt>
        <dd>使用ldsc软件进行临床性状遗传力</dd>
      </dl><dl><dt>

          <code><a href="LDSC_py_ph.html">LDSC_py_ph()</a></code>

        </dt>
        <dd>使用ldsc软件进行分区遗传力计算</dd>
      </dl><dl><dt>

          <code><a href="LDSC_py_rg.html">LDSC_py_rg()</a></code>

        </dt>
        <dd>使用ldsc软件进行遗传相关性分析</dd>
      </dl><dl><dt>

          <code><a href="LDSC_py_SEG.html">LDSC_py_SEG()</a></code>

        </dt>
        <dd>使用ldsc软件进行组织或细胞类型特异性分析</dd>
      </dl></div><div class="section level2">

      <h3 id="-hdl-">13.HDL分析<a class="anchor" aria-label="anchor" href="#-hdl-"></a></h3>



    </div><div class="section level2">




      <dl><dt>

          <code><a href="HDL_h2.html">HDL_h2()</a></code>

        </dt>
        <dd>使用HDL包基于SNP的遗传力的高精度似然推断</dd>
      </dl><dl><dt>

          <code><a href="HDL_rg.html">HDL_rg()</a></code>

        </dt>
        <dd>使用HDL进行高精度似然推断遗传相关性分析</dd>
      </dl><dl><dt>

          <code><a href="HDL_rg_parallel.html">HDL_rg_parallel()</a></code>

        </dt>
        <dd>使用HDL包多核进行遗传相关性的高精度似然推断</dd>
      </dl><dl><dt>

          <code><a href="HDL_L.html">HDL_L()</a></code>

        </dt>
        <dd>使用HDL包进行局部遗传相关性分析</dd>
      </dl></div><div class="section level2">

      <h3 id="-lava-">14.LAVA分析<a class="anchor" aria-label="anchor" href="#-lava-"></a></h3>



    </div><div class="section level2">




      <dl><dt>

          <code><a href="LAVA_sample_overlap.html">LAVA_sample_overlap()</a></code>

        </dt>
        <dd>创建 LAVA 分析输入的sample overlap文件</dd>
      </dl><dl><dt>

          <code><a href="LAVA_run.html">LAVA_run()</a></code>

        </dt>
        <dd>使用 LAVA 包对所有基因组位点执行双变量遗传相关性分析</dd>
      </dl><dl><dt>

          <code><a href="LAVA_bivarplot.html">LAVA_bivarplot()</a></code>

        </dt>
        <dd>绘制双变量LAVA分析结果的散点图</dd>
      </dl></div><div class="section level2">

      <h3 id="-gnova-">15.GNOVA分析<a class="anchor" aria-label="anchor" href="#-gnova-"></a></h3>



    </div><div class="section level2">




      <dl><dt>

          <code><a href="GNOVA_py_install.html">GNOVA_py_install()</a></code>

        </dt>
        <dd>配置GNOVA软件环境</dd>
      </dl><dl><dt>

          <code><a href="GNOVA_run.html">GNOVA_run()</a></code>

        </dt>
        <dd>使用GNOVA进行遗传协方差分析</dd>
      </dl></div><div class="section level2">

      <h3 id="-supergnova-">16.SUPERGNOVA分析<a class="anchor" aria-label="anchor" href="#-supergnova-"></a></h3>



    </div><div class="section level2">




      <dl><dt>

          <code><a href="SUPERGNOVA_py_install.html">SUPERGNOVA_py_install()</a></code>

        </dt>
        <dd>配置SUPERGNOVA软件环境</dd>
      </dl><dl><dt>

          <code><a href="SUPERGNOVA_format_data.html">SUPERGNOVA_format_data()</a></code>

        </dt>
        <dd>将TwosampleMR格式转换成SUPERGNOVA分析输入格式数据</dd>
      </dl><dl><dt>

          <code><a href="SUPERGNOVA_run.html">SUPERGNOVA_run()</a></code>

        </dt>
        <dd>使用 SUPERGNOVA 进行局部遗传相关性分析</dd>
      </dl></div><div class="section level2">

      <h3 id="-hess-">17.HESS分析<a class="anchor" aria-label="anchor" href="#-hess-"></a></h3>



    </div><div class="section level2">




      <dl><dt>

          <code><a href="HESS_install.html">HESS_install()</a></code>

        </dt>
        <dd>配置HESS软件环境</dd>
      </dl><dl><dt>

          <code><a href="HESS_format_data.html">HESS_format_data()</a></code>

        </dt>
        <dd>将TwosampleMR格式转换成HESS分析输入格式数据</dd>
      </dl><dl><dt>

          <code><a href="HESS_estimate_heritability.html">HESS_estimate_heritability()</a></code>

        </dt>
        <dd>使用HESS软件计算单一性状局部SNP遗传力及标准误差</dd>
      </dl><dl><dt>

          <code><a href="HESS_estimate_correlation.html">HESS_estimate_correlation()</a></code>

        </dt>
        <dd>使用HESS软件计算性状遗传协方差和相关性(ρ-HESS)</dd>
      </dl><dl><dt>

          <code><a href="HESS_estimate_phenocor.html">HESS_estimate_phenocor()</a></code>

        </dt>
        <dd>使用HESS软件计算性状表型的相关性</dd>
      </dl><dl><dt>

          <code><a href="HESS_estimate_lambdagc.html">HESS_estimate_lambdagc()</a></code>

        </dt>
        <dd>使用HESS软件计算基因组控制因子</dd>
      </dl><dl><dt>

          <code><a href="HESS_Visualizing_estimates.html">HESS_Visualizing_estimates()</a></code>

        </dt>
        <dd>使用HESS软件绘制SNP遗传度或协方差的曼哈顿图</dd>
      </dl><dl><dt>

          <code><a href="HESS_putative_causality.html">HESS_putative_causality()</a></code>

        </dt>
        <dd>使用HESS软件推断性状特征之间因果关系并绘图</dd>
      </dl><dl><dt>

          <code><a href="HESS_Contrast_polygenicity.html">HESS_Contrast_polygenicity()</a></code>

        </dt>
        <dd>使用HESS软件绘制性状之间多基因性对比度的图</dd>
      </dl></div><div class="section level2">

      <h3 id="-mixer-linux-">18.MiXeR分析(Linux系统)<a class="anchor" aria-label="anchor" href="#-mixer-linux-"></a></h3>



    </div><div class="section level2">




      <dl><dt>

          <code><a href="MiXeR_install.html">MiXeR_install()</a></code>

        </dt>
        <dd>配置MiXeR软件环境</dd>
      </dl><dl><dt>

          <code><a href="MiXeR_format_dat.html">MiXeR_format_dat()</a></code>

        </dt>
        <dd>MiXeR分析输入数据格式转换</dd>
      </dl><dl><dt>

          <code><a href="MiXeR_Univariate.html">MiXeR_Univariate()</a></code>

        </dt>
        <dd>MiXeR单变量分析</dd>
      </dl><dl><dt>

          <code><a href="MiXeR_Univariate_visualize.html">MiXeR_Univariate_visualize()</a></code>

        </dt>
        <dd>MiXeR单变量分析可视化</dd>
      </dl><dl><dt>

          <code><a href="MiXeR_Bivariate.html">MiXeR_Bivariate()</a></code>

        </dt>
        <dd>MiXeR双变量跨性状分析</dd>
      </dl><dl><dt>

          <code><a href="MiXeR_Bivariate_visualize.html">MiXeR_Bivariate_visualize()</a></code>

        </dt>
        <dd>MiXeR双变量跨性状分析可视化</dd>
      </dl><dl><dt>

          <code><a href="MiXeR_run.html">MiXeR_run()</a></code>

        </dt>
        <dd>MiXeR分析</dd>
      </dl></div><div class="section level2">

      <h3 id="-fusion-">19.FUSION分析<a class="anchor" aria-label="anchor" href="#-fusion-"></a></h3>



    </div><div class="section level2">




      <dl><dt>

          <code><a href="TWAS_fusion_format_data.html">TWAS_fusion_format_data()</a></code>

        </dt>
        <dd>将TwosampleMR格式转换成TWAS Fusion分析输入格式数据</dd>
      </dl><dl><dt>

          <code><a href="TWAS_fusion_assoc_test.html">TWAS_fusion_assoc_test()</a></code>

        </dt>
        <dd>使用FUSION软件进行TWAS分析</dd>
      </dl><dl><dt>

          <code><a href="TWAS_fusion_Multi_test.html">TWAS_fusion_Multi_test()</a></code>

        </dt>
        <dd>使用FUSION软件进行循环多种组织分析</dd>
      </dl><dl><dt>

          <code><a href="TWAS_fusion_conditonal_test.html">TWAS_fusion_conditonal_test()</a></code>

        </dt>
        <dd>使用FUSION软件进行Joint/conditional分析</dd>
      </dl></div><div class="section level2">

      <h3 id="-utmost-">20.UTMOST分析<a class="anchor" aria-label="anchor" href="#-utmost-"></a></h3>



    </div><div class="section level2">




      <dl><dt>

          <code><a href="TWAS_UTMOST_install.html">TWAS_UTMOST_install()</a></code>

        </dt>
        <dd>配置UTMOST软件环境</dd>
      </dl><dl><dt>

          <code><a href="TWAS_UTMOST_single_tissue_test.html">TWAS_UTMOST_single_tissue_test()</a></code>

        </dt>
        <dd>使用UTMOST软件进行单一组织TWAS分析</dd>
      </dl><dl><dt>

          <code><a href="TWAS_UTMOST_cross_tissue_test.html">TWAS_UTMOST_cross_tissue_test()</a></code>

        </dt>
        <dd>使用UTMOST软件进行跨组织联合TWAS分析</dd>
      </dl></div><div class="section level2">

      <h3 id="-mr-jti-">21.MR-JTI分析<a class="anchor" aria-label="anchor" href="#-mr-jti-"></a></h3>



    </div><div class="section level2">




      <dl><dt>

          <code><a href="MetaXcan_install.html">MetaXcan_install()</a></code>

        </dt>
        <dd>配置MetaXcan软件环境</dd>
      </dl><dl><dt>

          <code><a href="JTI_predixcan_test.html">JTI_predixcan_test()</a></code>

        </dt>
        <dd>使用predixcan软件进行TWAS分析</dd>
      </dl><dl><dt>

          <code><a href="JTI_SPrediXcan_test.html">JTI_SPrediXcan_test()</a></code>

        </dt>
        <dd>使用S-PrediXcan软件进行TWAS分析</dd>
      </dl><dl><dt>

          <code><a href="MR_JTI_Beta.html">MR_JTI_Beta()</a></code>

        </dt>
        <dd>计算MR_JTI结果的beta值及95% CI</dd>
      </dl></div><div class="section level2">

      <h3 id="-smultixcan-">22.SMulTiXcan分析<a class="anchor" aria-label="anchor" href="#-smultixcan-"></a></h3>



    </div><div class="section level2">




      <dl><dt>

          <code><a href="MetaXcan_Harmonization.html">MetaXcan_Harmonization()</a></code>

        </dt>
        <dd>对 GWAS 基因组版本转换以适配 MetaXcan 分析</dd>
      </dl><dl><dt>

          <code><a href="MetaXcan_imputation.html">MetaXcan_imputation()</a></code>

        </dt>
        <dd>对 GWAS 结果的基因型插补（Imputation）以适配 MetaXcan 分析</dd>
      </dl><dl><dt>

          <code><a href="SPrediXcan.html">SPrediXcan()</a></code>

        </dt>
        <dd>使用GTEX v8 MAHR 预测模型进行S-PrediXcan分析</dd>
      </dl><dl><dt>

          <code><a href="SMulTiXcan.html">SMulTiXcan()</a></code>

        </dt>
        <dd>使用SMulTiXcan进行跨组织TWAS分析</dd>
      </dl></div><div class="section level2">

      <h3 id="-focus-">23.FOCUS分析<a class="anchor" aria-label="anchor" href="#-focus-"></a></h3>



    </div><div class="section level2">




      <dl><dt>

          <code><a href="TWAS_FOCUS_install.html">TWAS_FOCUS_install()</a></code>

        </dt>
        <dd>配置FOCUS软件环境</dd>
      </dl><dl><dt>

          <code><a href="TWAS_FOCUS_import.html">TWAS_FOCUS_import()</a></code>

        </dt>
        <dd>使用FOCUS软件创建权重数据</dd>
      </dl><dl><dt>

          <code><a href="TWAS_FOCUS_format_data.html">TWAS_FOCUS_format_data()</a></code>

        </dt>
        <dd>将TwosampleMR格式转换成TWAS FOCUS分析输入格式数据</dd>
      </dl><dl><dt>

          <code><a href="TWAS_FOCUS_test.html">TWAS_FOCUS_test()</a></code>

        </dt>
        <dd>使用FOCUS软件进行TWAS的基因-性状关联信号精细定位分析</dd>
      </dl><dl><dt>

          <code><a href="TWAS_FOCUS_Multi_test.html">TWAS_FOCUS_Multi_test()</a></code>

        </dt>
        <dd>使用FOCUS软件进行循环多种组织TWAS的基因-性状关联信号精细定位分析</dd>
      </dl></div><div class="section level2">

      <h3 id="-magma-">24.MAGMA分析<a class="anchor" aria-label="anchor" href="#-magma-"></a></h3>



    </div><div class="section level2">




      <dl><dt>

          <code><a href="MAGMA_gene_based.html">MAGMA_gene_based()</a></code>

        </dt>
        <dd>使用MAGMA软件进行gene-based &amp; gene-set-based关联分析</dd>
      </dl><dl><dt>

          <code><a href="MAGMA_Tissue_specific.html">MAGMA_Tissue_specific()</a></code>

        </dt>
        <dd>使用MAGMA进行交叉组织特异性分析</dd>
      </dl><dl><dt>

          <code><a href="MAGMA_genes_Manhattanplot.html">MAGMA_genes_Manhattanplot()</a></code>

        </dt>
        <dd>对MAGMA软件gene_based分析结果的基因注释及曼哈顿图绘制</dd>
      </dl><dl><dt>

          <code><a href="MAGMA_gsa_barplot.html">MAGMA_gsa_barplot()</a></code>

        </dt>
        <dd>绘制MAGMA分析结果柱状图</dd>
      </dl></div><div class="section level2">

      <h3 id="-pops-">25.PoPS分析<a class="anchor" aria-label="anchor" href="#-pops-"></a></h3>



    </div><div class="section level2">




      <dl><dt>

          <code><a href="PoPS_py_install.html">PoPS_py_install()</a></code>

        </dt>
        <dd>配置PoPS软件环境</dd>
      </dl><dl><dt>

          <code><a href="PoPS_munge_feature.html">PoPS_munge_feature()</a></code>

        </dt>
        <dd>PoPS特征预处理函数</dd>
      </dl><dl><dt>

          <code><a href="PoPS_MAGMA.html">PoPS_MAGMA()</a></code>

        </dt>
        <dd>使用MAGMA软件生成MAGMA分数</dd>
      </dl><dl><dt>

          <code><a href="PoPS_py_run.html">PoPS_py_run()</a></code>

        </dt>
        <dd>使用PoPS进行基因优先评分分析</dd>
      </dl><dl><dt>

          <code><a href="PoPS_run.html">PoPS_run()</a></code>

        </dt>
        <dd>PoPS分析</dd>
      </dl></div><div class="section level2">

      <h3 id="-gcta-">26.GCTA分析<a class="anchor" aria-label="anchor" href="#-gcta-"></a></h3>



    </div><div class="section level2">




      <dl><dt>

          <code><a href="GCTA_fastBAT.html">GCTA_fastBAT()</a></code>

        </dt>
        <dd>基于GCTA_fastBAT的gene-based关联分析研究</dd>
      </dl></div><div class="section level2">

      <h3 id="-gwas-meta-">27.GWAS-meta分析<a class="anchor" aria-label="anchor" href="#-gwas-meta-"></a></h3>



    </div><div class="section level2">




      <dl><dt>

          <code><a href="MTAG_install.html">MTAG_install()</a></code>

        </dt>
        <dd>配置MTAG软件环境</dd>
      </dl><dl><dt>

          <code><a href="MTAG_cross_gwas.html">MTAG_cross_gwas()</a></code>

        </dt>
        <dd>使用MTAG进行跨性状表型meta分析</dd>
      </dl><dl><dt>

          <code><a href="METAL_gwas.html">METAL_gwas()</a></code>

        </dt>
        <dd>使用METAL软件对GWAS进行meta分析</dd>
      </dl><dl><dt>

          <code><a href="CPASSOC_meta_gwas.html">CPASSOC_meta_gwas()</a></code>

        </dt>
        <dd>使用cpassoc跨性状关联分析</dd>
      </dl><dl><dt>

          <code><a href="cond_conjFDR.html">cond_conjFDR()</a></code>

        </dt>
        <dd>cond/conjFDR分析</dd>
      </dl><dl><dt>

          <code><a href="PLACO_analysis.html">PLACO_analysis()</a></code>

        </dt>
        <dd>使用PLACO进行复合零假设下的多效性分析</dd>
      </dl></div><div class="section level2">

      <h3 id="-fm-summary-">28.FM_summary分析<a class="anchor" aria-label="anchor" href="#-fm-summary-"></a></h3>



    </div><div class="section level2">




      <dl><dt>

          <code><a href="FM_summary.html">FM_summary()</a></code>

        </dt>
        <dd>简化的贝叶斯精细定位方法（FM-summary）</dd>
      </dl></div><div class="section level2">

      <h3 id="--5">29.其它分析功能<a class="anchor" aria-label="anchor" href="#--5"></a></h3>



    </div><div class="section level2">




      <dl><dt>

          <code><a href="LCV_run.html">LCV_run()</a></code>

        </dt>
        <dd>潜在因果变量分析(LCV)</dd>
      </dl><dl><dt>

          <code><a href="find_Confounders_ensembl.html">find_Confounders_ensembl()</a></code>

        </dt>
        <dd>从ensembl数据库获取混杂因素</dd>
      </dl><dl><dt>

          <code><a href="MRPRESSO_outliers_snp.html">MRPRESSO_outliers_snp()</a></code>

        </dt>
        <dd>MR-PRESSO逐一剔除水平多效性的SNP</dd>
      </dl><dl><dt>

          <code><a href="clump_LD_R2.html">clump_LD_R2()</a></code>

        </dt>
        <dd>获取连锁的SNP</dd>
      </dl><dl><dt>

          <code><a href="get_EAF_local.html">get_EAF_local()</a></code>

        </dt>
        <dd>本地文件获取EAF值（不推荐）</dd>
      </dl><dl><dt>

          <code><a href="venn_plot.html">venn_plot()</a></code>

        </dt>
        <dd>绘制韦恩图</dd>
      </dl><dl><dt>

          <code><a href="Visualizing_MR_forest.html">Visualizing_MR_forest()</a></code>

        </dt>
        <dd>绘制MR分析森林图</dd>
      </dl><dl><dt>

          <code><a href="Visualizing_Manhattan.html">Visualizing_Manhattan()</a></code>

        </dt>
        <dd>绘制曼哈顿图</dd>
      </dl><dl><dt>

          <code><a href="Visualizing_bidir_plot.html">Visualizing_bidir_plot()</a></code>

        </dt>
        <dd>绘制双向柱状图</dd>
      </dl></div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by YB H.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer></div>





  </body></html>

