<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>使用predixcan软件进行TWAS分析 — JTI_predixcan_test • QTLMR</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="使用predixcan软件进行TWAS分析 — JTI_predixcan_test"><meta name="description" content="软件官网：https://github.com/hakyimlab/MetaXcan/tree/master"><meta property="og:description" content="软件官网：https://github.com/hakyimlab/MetaXcan/tree/master"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">QTLMR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">2.14</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../index.html">主页</a></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">函数功能</a></li>
<li class="nav-item"><a class="nav-link" href="../articles/index.html">文章示例代码</a></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">更新日志</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>使用predixcan软件进行TWAS分析</h1>

      <div class="d-none name"><code>JTI_predixcan_test.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>软件官网：https://github.com/hakyimlab/MetaXcan/tree/master</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">JTI_predixcan_test</span><span class="op">(</span></span>
<span>  test_help <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  GTEX_tissue <span class="op">=</span> <span class="st">"all"</span>,</span>
<span>  model_db_path <span class="op">=</span> <span class="st">"./JTI"</span>,</span>
<span>  covariance <span class="op">=</span> <span class="st">"./JTI"</span>,</span>
<span>  gwas_file <span class="op">=</span> <span class="st">"SNP_gwas_mc_merge_nogc.tbl.uniq"</span>,</span>
<span>  snp_column <span class="op">=</span> <span class="st">"SNP"</span>,</span>
<span>  effect_allele_column <span class="op">=</span> <span class="st">"A1"</span>,</span>
<span>  non_effect_allele_column <span class="op">=</span> <span class="st">"A2"</span>,</span>
<span>  beta_column <span class="op">=</span> <span class="st">"b"</span>,</span>
<span>  se_column <span class="op">=</span> <span class="st">"se"</span>,</span>
<span>  opt_arguments <span class="op">=</span> <span class="st">"--parallel"</span>,</span>
<span>  FDR_method <span class="op">=</span> <span class="st">"bonferroni"</span>,</span>
<span>  save_path <span class="op">=</span> <span class="st">"./"</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-test-help">test_help<a class="anchor" aria-label="anchor" href="#arg-test-help"></a></dt>
<dd><p>是否调阅predixcan.py的帮助文档，默认FALSE。</p></dd>


<dt id="arg-gtex-tissue">GTEX_tissue<a class="anchor" aria-label="anchor" href="#arg-gtex-tissue"></a></dt>
<dd><p>GTEx v8组织类型，默认"all"，即对所有组织分析，也可指定组织，如"JTI_Whole_Blood"。</p></dd>


<dt id="arg-model-db-path">model_db_path<a class="anchor" aria-label="anchor" href="#arg-model-db-path"></a></dt>
<dd><p>基因表达补全模型的路径（顺式eQTLs的估计权重/效应值）。</p></dd>


<dt id="arg-covariance">covariance<a class="anchor" aria-label="anchor" href="#arg-covariance"></a></dt>
<dd><p>包含协方差信息的文件路径（用于估计基因水平效应量估计器的方差，详情见手稿方法部分的基因水平关联测试）。</p></dd>


<dt id="arg-gwas-file">gwas_file<a class="anchor" aria-label="anchor" href="#arg-gwas-file"></a></dt>
<dd><p>GWAS的文件路径。</p></dd>


<dt id="arg-snp-column">snp_column<a class="anchor" aria-label="anchor" href="#arg-snp-column"></a></dt>
<dd><p>包含RSID的列名。</p></dd>


<dt id="arg-effect-allele-column">effect_allele_column<a class="anchor" aria-label="anchor" href="#arg-effect-allele-column"></a></dt>
<dd><p>包含效应等位基因的列名。</p></dd>


<dt id="arg-non-effect-allele-column">non_effect_allele_column<a class="anchor" aria-label="anchor" href="#arg-non-effect-allele-column"></a></dt>
<dd><p>包含非效应等位基因的列名。</p></dd>


<dt id="arg-beta-column">beta_column<a class="anchor" aria-label="anchor" href="#arg-beta-column"></a></dt>
<dd><p>输入GWAS文件中包含每个SNP效应值的列名。</p></dd>


<dt id="arg-se-column">se_column<a class="anchor" aria-label="anchor" href="#arg-se-column"></a></dt>
<dd><p>输入GWAS文件中包含每个SNP 标准方差值的列名。</p></dd>


<dt id="arg-opt-arguments">opt_arguments<a class="anchor" aria-label="anchor" href="#arg-opt-arguments"></a></dt>
<dd><p>predixcan.py的其他命令行参数(设置test_help=TRUE查看)，默认NULL。</p></dd>


<dt id="arg-fdr-method">FDR_method<a class="anchor" aria-label="anchor" href="#arg-fdr-method"></a></dt>
<dd><p>对P值进行FDR矫正方法，"fdr"或"bonferroni"，默认"bonferroni"。</p></dd>


<dt id="arg-save-path">save_path<a class="anchor" aria-label="anchor" href="#arg-save-path"></a></dt>
<dd><p>文件保存路径。</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>分析结果。</p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>目的是通过结合基因组关联研究（GWAS）的汇总统计数据和基因表达数据，来预测和测试基因表达与特定疾病或表型之间的相关性。</p>
    </div>

  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by H YB.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer></div>





  </body></html>

