<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>使用METAL软件对GWAS进行meta分析 — METAL_gwas • QTLMR</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="使用METAL软件对GWAS进行meta分析 — METAL_gwas"><meta name="description" content="windows系统下使用METAL软件对全基因组关联研究(GWAS)数据进行META分析，电脑内存小容易运行崩溃。"><meta property="og:description" content="windows系统下使用METAL软件对全基因组关联研究(GWAS)数据进行META分析，电脑内存小容易运行崩溃。"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">QTLMR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.82</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../index.html">主页</a></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">函数功能</a></li>
<li class="nav-item"><a class="nav-link" href="../articles/index.html">文章示例代码</a></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">更新日志</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>使用METAL软件对GWAS进行meta分析</h1>

      <div class="d-none name"><code>METAL_gwas.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>windows系统下使用METAL软件对全基因组关联研究(GWAS)数据进行META分析，电脑内存小容易运行崩溃。</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">METAL_gwas</span><span class="op">(</span></span>
<span>  GWASfile <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"./dat1.txt"</span>, <span class="st">"./dat1.txt"</span><span class="op">)</span>,</span>
<span>  SEPARATOR <span class="op">=</span> <span class="st">"TAB"</span>,</span>
<span>  snp_col <span class="op">=</span> <span class="st">"SNP"</span>,</span>
<span>  effect_allele_col <span class="op">=</span> <span class="st">"effect_allele"</span>,</span>
<span>  other_allele_col <span class="op">=</span> <span class="st">"other_allele"</span>,</span>
<span>  beta_col <span class="op">=</span> <span class="st">"beta"</span>,</span>
<span>  se_col <span class="op">=</span> <span class="st">"se"</span>,</span>
<span>  pval_col <span class="op">=</span> <span class="st">"pval"</span>,</span>
<span>  eaf_col <span class="op">=</span> <span class="st">"eaf"</span>,</span>
<span>  samplesize_col <span class="op">=</span> <span class="st">"N"</span>,</span>
<span>  chr_col <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  pos_col <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  analysis_method <span class="op">=</span> <span class="st">"SE"</span>,</span>
<span>  IVW_mode <span class="op">=</span> <span class="st">"Multiplicative"</span>,</span>
<span>  Genomic_Control_Correction <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  Average_FREQ <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  Heterogeneity <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  Sample_Overlap_Correction <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  OVERLAP_ZCUTOFF <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  Filter_MAF <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  Filter_N <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  Filter_SNPs <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  save_path <span class="op">=</span> <span class="st">"./"</span>,</span>
<span>  save_name <span class="op">=</span> <span class="st">"METAANALYSIS"</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-gwasfile">GWASfile<a class="anchor" aria-label="anchor" href="#arg-gwasfile"></a></dt>
<dd><p>包含两个或多个GWAS数据文件的路径。</p></dd>


<dt id="arg-separator">SEPARATOR<a class="anchor" aria-label="anchor" href="#arg-separator"></a></dt>
<dd><p>指定数据文件中列之间的分隔符，默认为"TAB"。</p></dd>


<dt id="arg-snp-col">snp_col<a class="anchor" aria-label="anchor" href="#arg-snp-col"></a></dt>
<dd><p>SNP列的名称。</p></dd>


<dt id="arg-effect-allele-col">effect_allele_col<a class="anchor" aria-label="anchor" href="#arg-effect-allele-col"></a></dt>
<dd><p>效应等位基因列的名称。</p></dd>


<dt id="arg-other-allele-col">other_allele_col<a class="anchor" aria-label="anchor" href="#arg-other-allele-col"></a></dt>
<dd><p>次要等位基因列的名称。</p></dd>


<dt id="arg-beta-col">beta_col<a class="anchor" aria-label="anchor" href="#arg-beta-col"></a></dt>
<dd><p>β值列的名称。</p></dd>


<dt id="arg-se-col">se_col<a class="anchor" aria-label="anchor" href="#arg-se-col"></a></dt>
<dd><p>标准误差列的名称。</p></dd>


<dt id="arg-pval-col">pval_col<a class="anchor" aria-label="anchor" href="#arg-pval-col"></a></dt>
<dd><p>P值列的名称。</p></dd>


<dt id="arg-eaf-col">eaf_col<a class="anchor" aria-label="anchor" href="#arg-eaf-col"></a></dt>
<dd><p>效应等位基因频率列的名称。</p></dd>


<dt id="arg-samplesize-col">samplesize_col<a class="anchor" aria-label="anchor" href="#arg-samplesize-col"></a></dt>
<dd><p>样本量列的名称。</p></dd>


<dt id="arg-chr-col">chr_col<a class="anchor" aria-label="anchor" href="#arg-chr-col"></a></dt>
<dd><p>染色体列的名称，默认为NULL，输入数据为GRCH38(hg38)或GRCH37(hg19)相同版本时填写，否则合并错误。</p></dd>


<dt id="arg-pos-col">pos_col<a class="anchor" aria-label="anchor" href="#arg-pos-col"></a></dt>
<dd><p>坐标列的名称，默认为NULL，输入数据为GRCH38(hg38)或GRCH37(hg19)相同版本时填写，否则合并错误。</p></dd>


<dt id="arg-analysis-method">analysis_method<a class="anchor" aria-label="anchor" href="#arg-analysis-method"></a></dt>
<dd><p>"SE"或"samplesize"，SE表示逆方差加权法-IVW，samplesize表示基于样本量加权。</p></dd>


<dt id="arg-ivw-mode">IVW_mode<a class="anchor" aria-label="anchor" href="#arg-ivw-mode"></a></dt>
<dd><p>逆方差加权法-IVW分析模式，固定效应模型:"Fixed"，随机效应模型:"Multiplicative"或"Additive"。</p></dd>


<dt id="arg-genomic-control-correction">Genomic_Control_Correction<a class="anchor" aria-label="anchor" href="#arg-genomic-control-correction"></a></dt>
<dd><p>逻辑值，是否进行基因组控制校正，默认为TRUE。</p></dd>


<dt id="arg-average-freq">Average_FREQ<a class="anchor" aria-label="anchor" href="#arg-average-freq"></a></dt>
<dd><p>TRUE或FALSE，是否计算平均频率，默认为TRUE。</p></dd>


<dt id="arg-heterogeneity">Heterogeneity<a class="anchor" aria-label="anchor" href="#arg-heterogeneity"></a></dt>
<dd><p>TRUE或FALSE，是否进行异质性分析，默认为TRUE。</p></dd>


<dt id="arg-sample-overlap-correction">Sample_Overlap_Correction<a class="anchor" aria-label="anchor" href="#arg-sample-overlap-correction"></a></dt>
<dd><p>TRUE或FALSE，是否进行样本重叠校正，默认为TRUE。</p></dd>


<dt id="arg-overlap-zcutoff">OVERLAP_ZCUTOFF<a class="anchor" aria-label="anchor" href="#arg-overlap-zcutoff"></a></dt>
<dd><p>数值，样本重叠校正的Z截断值，默认为1。</p></dd>


<dt id="arg-filter-maf">Filter_MAF<a class="anchor" aria-label="anchor" href="#arg-filter-maf"></a></dt>
<dd><p>数值，次要等位基因频率过滤值，默认为NULL。</p></dd>


<dt id="arg-filter-n">Filter_N<a class="anchor" aria-label="anchor" href="#arg-filter-n"></a></dt>
<dd><p>数值，样本大小过滤值，默认为NULL。</p></dd>


<dt id="arg-filter-snps">Filter_SNPs<a class="anchor" aria-label="anchor" href="#arg-filter-snps"></a></dt>
<dd><p>字符向量，需要过滤的SNP，默认为NULL。</p></dd>


<dt id="arg-save-path">save_path<a class="anchor" aria-label="anchor" href="#arg-save-path"></a></dt>
<dd><p>分析结果保存路径，默认为"./"。</p></dd>


<dt id="arg-save-name">save_name<a class="anchor" aria-label="anchor" href="#arg-save-name"></a></dt>
<dd><p>分析结果文件名，默认为"METAANALYSIS"。</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>无返回值，将执行GWAS分析并将结果保存到指定路径。</p>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by H YB.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer></div>





  </body></html>

