<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>对 GWAS 基因组版本转换以适配 MetaXcan 分析 — MetaXcan_Harmonization • QTLMR</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="对 GWAS 基因组版本转换以适配 MetaXcan 分析 — MetaXcan_Harmonization"><meta name="description" content="该函数用于对 GWAS 结果文件进行标准化处理，包括基因组坐标转换、SNP 信息对齐以及格式调整，以适配 MetaXcan 工具进行后续分析，
详见：https://github.com/hakyimlab/MetaXcan/wiki/Tutorial:-GTEx-v8-MASH-models-integration-with-a-Coronary-Artery-Disease-GWAS"><meta property="og:description" content="该函数用于对 GWAS 结果文件进行标准化处理，包括基因组坐标转换、SNP 信息对齐以及格式调整，以适配 MetaXcan 工具进行后续分析，
详见：https://github.com/hakyimlab/MetaXcan/wiki/Tutorial:-GTEx-v8-MASH-models-integration-with-a-Coronary-Artery-Disease-GWAS"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">QTLMR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">2.41</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../index.html">主页</a></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">函数功能</a></li>
<li class="nav-item"><a class="nav-link" href="../articles/index.html">文章示例代码</a></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">更新日志</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>对 GWAS 基因组版本转换以适配 MetaXcan 分析</h1>

      <div class="d-none name"><code>MetaXcan_Harmonization.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>该函数用于对 GWAS 结果文件进行标准化处理，包括基因组坐标转换、SNP 信息对齐以及格式调整，以适配 MetaXcan 工具进行后续分析，
详见：https://github.com/hakyimlab/MetaXcan/wiki/Tutorial:-GTEx-v8-MASH-models-integration-with-a-Coronary-Artery-Disease-GWAS</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">MetaXcan_Harmonization</span><span class="op">(</span></span>
<span>  gwas_file <span class="op">=</span> <span class="st">"./XXX_MTAG.txt"</span>,</span>
<span>  liftover <span class="op">=</span> <span class="st">"./hg19ToHg38.over.chain.gz"</span>,</span>
<span>  snp_reference_metadata <span class="op">=</span> <span class="st">"./reference_panel_1000G/variant_metadata.txt.gz"</span>,</span>
<span>  snp_col <span class="op">=</span> <span class="st">"SNP"</span>,</span>
<span>  chr_col <span class="op">=</span> <span class="st">"CHR"</span>,</span>
<span>  pos_col <span class="op">=</span> <span class="st">"BP"</span>,</span>
<span>  beta_col <span class="op">=</span> <span class="st">"beta"</span>,</span>
<span>  se_col <span class="op">=</span> <span class="st">"se"</span>,</span>
<span>  eaf_col <span class="op">=</span> <span class="st">"eaf"</span>,</span>
<span>  effect_allele_col <span class="op">=</span> <span class="st">"effect_allele"</span>,</span>
<span>  other_allele_col <span class="op">=</span> <span class="st">"other_allele"</span>,</span>
<span>  pval_col <span class="op">=</span> <span class="st">"pval"</span>,</span>
<span>  samplesize_col <span class="op">=</span> <span class="st">"N"</span>,</span>
<span>  ncase <span class="op">=</span> <span class="fl">5000</span>,</span>
<span>  opt_arguments <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  save_name <span class="op">=</span> <span class="st">"bmi_MTAG"</span>,</span>
<span>  save_path <span class="op">=</span> <span class="st">"./"</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-gwas-file">gwas_file<a class="anchor" aria-label="anchor" href="#arg-gwas-file"></a></dt>
<dd><p>字符串，GWAS 结果文件的路径（默认为"./XXX_MTAG.txt"）。</p></dd>


<dt id="arg-liftover">liftover<a class="anchor" aria-label="anchor" href="#arg-liftover"></a></dt>
<dd><p>字符串，基因组坐标转换的链文件路径（默认为"./hg19ToHg38.over.chain.gz"）。</p></dd>


<dt id="arg-snp-reference-metadata">snp_reference_metadata<a class="anchor" aria-label="anchor" href="#arg-snp-reference-metadata"></a></dt>
<dd><p>字符串，SNP 参考数据文件路径。</p></dd>


<dt id="arg-snp-col">snp_col<a class="anchor" aria-label="anchor" href="#arg-snp-col"></a></dt>
<dd><p>字符串，GWAS 结果文件中 SNP 标识符的列名（默认 "SNP"）。</p></dd>


<dt id="arg-chr-col">chr_col<a class="anchor" aria-label="anchor" href="#arg-chr-col"></a></dt>
<dd><p>字符串，GWAS 结果文件中染色体编号的列名（默认 "CHR"）。</p></dd>


<dt id="arg-pos-col">pos_col<a class="anchor" aria-label="anchor" href="#arg-pos-col"></a></dt>
<dd><p>字符串，GWAS 结果文件中 SNP 物理位置（碱基对位置）的列名（默认 "BP"）。</p></dd>


<dt id="arg-beta-col">beta_col<a class="anchor" aria-label="anchor" href="#arg-beta-col"></a></dt>
<dd><p>字符串，GWAS 结果文件中效应值（回归系数 β）的列名（默认 "beta"）。</p></dd>


<dt id="arg-se-col">se_col<a class="anchor" aria-label="anchor" href="#arg-se-col"></a></dt>
<dd><p>字符串，GWAS 结果文件中标准误（standard error）的列名（默认 "se"）。</p></dd>


<dt id="arg-eaf-col">eaf_col<a class="anchor" aria-label="anchor" href="#arg-eaf-col"></a></dt>
<dd><p>字符串，GWAS 结果文件中效应等位基因频率（effect allele frequency, EAF）的列名（默认 "eaf"）。</p></dd>


<dt id="arg-effect-allele-col">effect_allele_col<a class="anchor" aria-label="anchor" href="#arg-effect-allele-col"></a></dt>
<dd><p>字符串，GWAS 结果文件中效应等位基因的列名（默认 "effect_allele"）。</p></dd>


<dt id="arg-other-allele-col">other_allele_col<a class="anchor" aria-label="anchor" href="#arg-other-allele-col"></a></dt>
<dd><p>字符串，GWAS 结果文件中非效应等位基因的列名（默认 "other_allele"）。</p></dd>


<dt id="arg-pval-col">pval_col<a class="anchor" aria-label="anchor" href="#arg-pval-col"></a></dt>
<dd><p>字符串，GWAS 结果文件中 P 值的列名（默认 "pval"）。</p></dd>


<dt id="arg-samplesize-col">samplesize_col<a class="anchor" aria-label="anchor" href="#arg-samplesize-col"></a></dt>
<dd><p>字符串，GWAS 结果文件中样本量的列名（默认 "N"）。</p></dd>


<dt id="arg-ncase">ncase<a class="anchor" aria-label="anchor" href="#arg-ncase"></a></dt>
<dd><p>数值，二分类GWAS数据中的病例数量；连续型GWAS数据设置为NULL。</p></dd>


<dt id="arg-opt-arguments">opt_arguments<a class="anchor" aria-label="anchor" href="#arg-opt-arguments"></a></dt>
<dd><p>字符串，额外的可选参数（默认为 NULL）。</p></dd>


<dt id="arg-save-name">save_name<a class="anchor" aria-label="anchor" href="#arg-save-name"></a></dt>
<dd><p>字符串，输出文件的名称。</p></dd>


<dt id="arg-save-path">save_path<a class="anchor" aria-label="anchor" href="#arg-save-path"></a></dt>
<dd><p>字符串，输出文件的存储路径（默认 "./"）。如果路径不存在，则会自动创建。</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>无返回值，函数将在指定路径生成标准化后的 GWAS 结果文件。</p>
    </div>

  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by YB H.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer></div>





  </body></html>

