<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>使用SMulTiXcan进行跨组织TWAS分析 — SMulTiXcan • QTLMR</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="使用SMulTiXcan进行跨组织TWAS分析 — SMulTiXcan"><meta name="description" content="该函数用于运行 SMulTiXcan 分析，以评估 GWAS 研究中基因表达对表型的影响，
详见：https://github.com/hakyimlab/MetaXcan/wiki/Tutorial:-GTEx-v8-MASH-models-integration-with-a-Coronary-Artery-Disease-GWAS"><meta property="og:description" content="该函数用于运行 SMulTiXcan 分析，以评估 GWAS 研究中基因表达对表型的影响，
详见：https://github.com/hakyimlab/MetaXcan/wiki/Tutorial:-GTEx-v8-MASH-models-integration-with-a-Coronary-Artery-Disease-GWAS"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">QTLMR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">2.34</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../index.html">主页</a></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">函数功能</a></li>
<li class="nav-item"><a class="nav-link" href="../articles/index.html">文章示例代码</a></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">更新日志</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>使用SMulTiXcan进行跨组织TWAS分析</h1>

      <div class="d-none name"><code>SMulTiXcan.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>该函数用于运行 SMulTiXcan 分析，以评估 GWAS 研究中基因表达对表型的影响，
详见：https://github.com/hakyimlab/MetaXcan/wiki/Tutorial:-GTEx-v8-MASH-models-integration-with-a-Coronary-Artery-Disease-GWAS</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">SMulTiXcan</span><span class="op">(</span></span>
<span>  test_help <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  models_folder <span class="op">=</span> <span class="st">"./data/models/eqtl/mashr"</span>,</span>
<span>  models_name_pattern <span class="op">=</span> <span class="st">"mashr_(.*).db"</span>,</span>
<span>  snp_covariance <span class="op">=</span></span>
<span>    <span class="st">"./data/models/gtex_v8_expression_mashr_snp_smultixcan_covariance.txt.gz"</span>,</span>
<span>  metaxcan_folder <span class="op">=</span> <span class="st">"./spredixcan"</span>,</span>
<span>  metaxcan_filter <span class="op">=</span> <span class="st">"bmi_MTAG__PM__(.*).csv"</span>,</span>
<span>  metaxcan_file_name_parse_pattern <span class="op">=</span> <span class="st">"(.*)__PM__mashr_(.*).csv"</span>,</span>
<span>  model_db_snp_key <span class="op">=</span> <span class="st">"varID"</span>,</span>
<span>  gwas_file <span class="op">=</span> <span class="st">"./imputation/imputed_bmi_MTAG.txt.gz"</span>,</span>
<span>  snp_col <span class="op">=</span> <span class="st">"panel_variant_id"</span>,</span>
<span>  effect_allele_col <span class="op">=</span> <span class="st">"effect_allele"</span>,</span>
<span>  other_allele_col <span class="op">=</span> <span class="st">"non_effect_allele"</span>,</span>
<span>  zscore_col <span class="op">=</span> <span class="st">"zscore"</span>,</span>
<span>  cutoff_threshold <span class="op">=</span> <span class="fl">30</span>,</span>
<span>  verbosity <span class="op">=</span> <span class="fl">7</span>,</span>
<span>  throw <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>  opt_arguments <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  save_name <span class="op">=</span> <span class="st">"BMI"</span>,</span>
<span>  save_path <span class="op">=</span> <span class="st">"./smultixcan"</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-test-help">test_help<a class="anchor" aria-label="anchor" href="#arg-test-help"></a></dt>
<dd><p>逻辑值，是否显示帮助信息（TRUE: 显示帮助，FALSE: 运行分析）。</p></dd>


<dt id="arg-models-folder">models_folder<a class="anchor" aria-label="anchor" href="#arg-models-folder"></a></dt>
<dd><p>字符串，预测模型所在的文件夹路径。</p></dd>


<dt id="arg-models-name-pattern">models_name_pattern<a class="anchor" aria-label="anchor" href="#arg-models-name-pattern"></a></dt>
<dd><p>字符串，正则表达式，用于匹配模型文件中的组织名称。</p></dd>


<dt id="arg-snp-covariance">snp_covariance<a class="anchor" aria-label="anchor" href="#arg-snp-covariance"></a></dt>
<dd><p>字符串，SNP 协方差矩阵文件的路径。</p></dd>


<dt id="arg-metaxcan-folder">metaxcan_folder<a class="anchor" aria-label="anchor" href="#arg-metaxcan-folder"></a></dt>
<dd><p>字符串，MetaXcan 结果文件所在文件夹的路径。如SPrediXcan()函数生成的结果文件的路径。</p></dd>


<dt id="arg-metaxcan-filter">metaxcan_filter<a class="anchor" aria-label="anchor" href="#arg-metaxcan-filter"></a></dt>
<dd><p>字符串，正则表达式，用于筛选 MetaXcan 结果文件。如SPrediXcan()函数生成的结果文件名的正则表达式。</p></dd>


<dt id="arg-metaxcan-file-name-parse-pattern">metaxcan_file_name_parse_pattern<a class="anchor" aria-label="anchor" href="#arg-metaxcan-file-name-parse-pattern"></a></dt>
<dd><p>字符串，正则表达式，用于解析 MetaXcan 结果文件中的表型名称和模型名称。使用mashr模型时默认为"(.<em>)_<em>PM__mashr</em>(.</em>).csv"。</p></dd>


<dt id="arg-model-db-snp-key">model_db_snp_key<a class="anchor" aria-label="anchor" href="#arg-model-db-snp-key"></a></dt>
<dd><p>字符串，指定用于 SNP ID 的关键字。</p></dd>


<dt id="arg-gwas-file">gwas_file<a class="anchor" aria-label="anchor" href="#arg-gwas-file"></a></dt>
<dd><p>字符串，GWAS 研究的输入文件路径。如MetaXcan_imputation()函数生成的结果文件的路径。</p></dd>


<dt id="arg-snp-col">snp_col<a class="anchor" aria-label="anchor" href="#arg-snp-col"></a></dt>
<dd><p>字符串，GWAS 文件中 rsid 对应的列名。</p></dd>


<dt id="arg-effect-allele-col">effect_allele_col<a class="anchor" aria-label="anchor" href="#arg-effect-allele-col"></a></dt>
<dd><p>字符串，GWAS 文件中效应等位基因的列名。</p></dd>


<dt id="arg-other-allele-col">other_allele_col<a class="anchor" aria-label="anchor" href="#arg-other-allele-col"></a></dt>
<dd><p>字符串，GWAS 文件中非效应等位基因的列名。</p></dd>


<dt id="arg-zscore-col">zscore_col<a class="anchor" aria-label="anchor" href="#arg-zscore-col"></a></dt>
<dd><p>字符串，GWAS 文件中 Z 评分对应的列名。</p></dd>


<dt id="arg-cutoff-threshold">cutoff_threshold<a class="anchor" aria-label="anchor" href="#arg-cutoff-threshold"></a></dt>
<dd><p>数值，奇异值分解（SVD）截断的方差阈值。</p></dd>


<dt id="arg-verbosity">verbosity<a class="anchor" aria-label="anchor" href="#arg-verbosity"></a></dt>
<dd><p>整数，日志详细程度（1: 详细日志，10: 仅高层日志，&gt;10: 几乎无日志）。</p></dd>


<dt id="arg-throw">throw<a class="anchor" aria-label="anchor" href="#arg-throw"></a></dt>
<dd><p>逻辑值，是否在遇到错误时抛出异常（TRUE: 抛出异常，FALSE: 继续执行）。</p></dd>


<dt id="arg-opt-arguments">opt_arguments<a class="anchor" aria-label="anchor" href="#arg-opt-arguments"></a></dt>
<dd><p>字符串，额外的命令行参数，默认为 NULL。</p></dd>


<dt id="arg-save-name">save_name<a class="anchor" aria-label="anchor" href="#arg-save-name"></a></dt>
<dd><p>字符串，输出文件的名称。</p></dd>


<dt id="arg-save-path">save_path<a class="anchor" aria-label="anchor" href="#arg-save-path"></a></dt>
<dd><p>字符串，输出文件的存储路径。</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>无返回值，函数会执行 MetaXcan 分析并生成结果文件。</p>
    </div>

  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by YB H.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer></div>





  </body></html>

