<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>1.跨组织转录组关联研究寻找疾病易感基因 • QTLMR</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="1.跨组织转录组关联研究寻找疾病易感基因">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">QTLMR</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">2.32</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../index.html">主页</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">函数功能</a></li>
<li class="active nav-item"><a class="nav-link" href="../articles/index.html">文章示例代码</a></li>
<li class="nav-item"><a class="nav-link" href="../news/index.html">更新日志</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>1.跨组织转录组关联研究寻找疾病易感基因</h1>
            
            <h4 data-toc-skip class="date">2024-07-26</h4>
      

      <div class="d-none name"><code>1.跨组织转录组关联研究寻找疾病易感基因.Rmd</code></div>
    </div>

    
    
<ul>
<li>复现文章：PMID:
38840241，因清洗数据差异，只能复现大部分结果保持一致！</li>
</ul>
<div class="section level2">
<h2 id="section"></h2>
<div class="section level4">
<h4 id="载入qtlmr包">
<strong>1. 载入QTLMR包</strong><a class="anchor" aria-label="anchor" href="#%E8%BD%BD%E5%85%A5qtlmr%E5%8C%85"></a>
</h4>
<pre><code><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://hortoner.github.io/QTLMR">QTLMR</a></span><span class="op">)</span></span></code></pre>
</div>
<div class="section level4">
<h4 id="下载数据">
<strong>2.下载数据</strong><a class="anchor" aria-label="anchor" href="#%E4%B8%8B%E8%BD%BD%E6%95%B0%E6%8D%AE"></a>
</h4>
<ul>
<li>获取FinnGen数据的信息</li>
</ul>
<pre><code><span><span class="va">data_info_finngen</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/data_info_finngen.html">data_info_finngen</a></span><span class="op">(</span><span class="op">)</span></span></code></pre>
<ul>
<li>从指定网址下载Migraine数据（偏头痛数据）</li>
<li>链接：<a href="https://storage.googleapis.com/finngen-public-data-r10/summary_stats/finngen_R10_G6_MIGRAINE.gz" class="external-link uri">https://storage.googleapis.com/finngen-public-data-r10/summary_stats/finngen_R10_G6_MIGRAINE.gz</a>
</li>
</ul>
<p><img src="images%2Fimg1%2F1.ifo.png" width="100%" style="display: block; margin: auto;"></p>
</div>
<div class="section level4">
<h4 id="进行utmost分析">
<strong>3.进行UTMOST分析</strong><a class="anchor" aria-label="anchor" href="#%E8%BF%9B%E8%A1%8Cutmost%E5%88%86%E6%9E%90"></a>
</h4>
<ul>
<li>3.1 数据转换：将GWAS文件转换为适合UTMOST分析的格式</li>
</ul>
<pre><code><span><span class="va">dat</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/format_data_FinnGen.html">format_data_FinnGen</a></span><span class="op">(</span></span>
<span>GWASfile <span class="op">=</span> <span class="st">"./1.data/finngen_R10_G6_MIGRAINE.gz"</span>,  <span class="co"># 输入的GWAS文件路径</span></span>
<span>GWAS_name <span class="op">=</span> <span class="st">"Migraine"</span>,  <span class="co"># GWAS的名称</span></span>
<span>save_path <span class="op">=</span> <span class="st">"./1.data"</span>,  <span class="co"># 转换后数据的保存路径</span></span>
<span>type <span class="op">=</span> <span class="st">"outcome"</span>,  <span class="co"># 数据类型，表示这是结果数据</span></span>
<span>Twosample_dat <span class="op">=</span> <span class="cn">TRUE</span>,  <span class="co"># 是否保存Twosample格式数据</span></span>
<span>SMR_dat <span class="op">=</span> <span class="cn">TRUE</span>,  <span class="co"># 是否保存SMR格式数据</span></span>
<span>METAL_dat <span class="op">=</span> <span class="cn">TRUE</span>  <span class="co"># 是否保存METAL格式数据</span></span>
<span><span class="op">)</span></span></code></pre>
<p><img src="images%2Fimg1%2F2.format_data.jpg" width="100%" style="display: block; margin: auto;"></p>
<ul>
<li>3.2 单组织分析：进行单个组织的TWAS-UTMOST分析</li>
</ul>
<pre><code><span><span class="va">data_utmost</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/TWAS_UTMOST_single_tissue_test.html">TWAS_UTMOST_single_tissue_test</a></span><span class="op">(</span></span>
<span>  test_help <span class="op">=</span> <span class="cn">FALSE</span>,  <span class="co"># 是否显示帮助文档信息</span></span>
<span>  GTEX_tissue <span class="op">=</span> <span class="st">"all"</span>,  <span class="co"># 使用所有组织数据</span></span>
<span>  model_db_path <span class="op">=</span> <span class="st">"./2.UTMOST/database_normalized_pruned"</span>,  <span class="co"># 归一化后的模型数据库路径</span></span>
<span>  covariance <span class="op">=</span> <span class="st">"./2.UTMOST/covariance_GTEx8_normalized_pruned"</span>,  <span class="co"># 归一化后的协方差矩阵路径</span></span>
<span>  gwas_folder <span class="op">=</span> <span class="st">"./1.data"</span>,  <span class="co"># GWAS数据的文件夹路径</span></span>
<span>  gwas_file_pattern <span class="op">=</span> <span class="st">"Migraine_METAL_FinnGen.txt"</span>,  <span class="co"># GWAS的文件名</span></span>
<span>  snp_column <span class="op">=</span> <span class="st">"SNP"</span>,  <span class="co"># SNP列的名称</span></span>
<span>  effect_allele_column <span class="op">=</span> <span class="st">"effect_allele"</span>,  <span class="co"># 效应等位基因列的名称</span></span>
<span>  non_effect_allele_column <span class="op">=</span> <span class="st">"other_allele"</span>,  <span class="co"># 非效应等位基因列的名称</span></span>
<span>  beta_column <span class="op">=</span> <span class="st">"beta"</span>,  <span class="co"># beta系数列的名称</span></span>
<span>  pvalue_column <span class="op">=</span> <span class="st">"pval"</span>,  <span class="co"># p值列的名称</span></span>
<span>  opt_arguments <span class="op">=</span> <span class="cn">NULL</span>,  <span class="co"># 额外的参数</span></span>
<span>  save_path <span class="op">=</span> <span class="st">"./2.UTMOST"</span>,  <span class="co"># 保存结果的路径</span></span>
<span>  cores <span class="op">=</span> <span class="fl">14</span>  <span class="co"># 运行使用电脑线程数</span></span>
<span><span class="op">)</span></span></code></pre>
<p><img src="images%2Fimg1%2F3.utmost_res.png" width="100%" style="display: block; margin: auto;"></p>
<ul>
<li>3.3 跨组织联合分析：进行跨组织的TWAS-UTMOST分析</li>
</ul>
<pre><code><span><span class="co">#在运行此步之前，需要删除"./2.UTMOST"文件夹下的“all.csv"文件</span></span>
<span><span class="va">data_GBJ</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/TWAS_UTMOST_cross_tissue_test.html">TWAS_UTMOST_cross_tissue_test</a></span><span class="op">(</span></span>
<span>  weight_db <span class="op">=</span> <span class="st">"./2.UTMOST/database_normalized_pruned"</span>,  <span class="co"># 权重数据的文件路径</span></span>
<span>  cov_dir <span class="op">=</span> <span class="st">"./3.UTMOST_GBJ/covariance_joint_GTEx8_normalized_pruned"</span>,  <span class="co"># 协方差矩阵的文件路径</span></span>
<span>  input_folder <span class="op">=</span> <span class="st">"./2.UTMOST"</span>,  <span class="co"># 输入文件夹路径</span></span>
<span>  gene_info <span class="op">=</span> <span class="st">"./2.UTMOST/gene_info.txt"</span>,  <span class="co"># 基因信息文件路径</span></span>
<span>  cores <span class="op">=</span> <span class="fl">2</span>,    <span class="co"># 运行使用电脑线程数</span></span>
<span>  output_dir <span class="op">=</span> <span class="st">"./3.UTMOST_GBJ"</span>,  <span class="co"># 输出目录路径</span></span>
<span>  output_name <span class="op">=</span> <span class="st">"Migraine"</span>  <span class="co"># 输出文件名</span></span>
<span><span class="op">)</span></span></code></pre>
<p><img src="images%2Fimg1%2F4.GBJ.jpg.jpg" width="100%" style="display: block; margin: auto;"></p>
<ul>
<li>3.4 对UTMOST分析结果进行整理保存</li>
</ul>
<pre><code><span><span class="co"># 对UTMOST分析结果进行处理</span></span>
<span><span class="va">dat_utmost_GBJ</span> <span class="op">&lt;-</span> <span class="va">data_GBJ</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">left_join</a></span><span class="op">(</span><span class="fu"><a href="../reference/data_info_GTEx_V8_GENE.html">data_info_GTEx_V8_GENE</a></span><span class="op">(</span><span class="op">)</span>, by <span class="op">=</span> <span class="st">"ID"</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span>  <span class="co"># 将结果与GTEx基因信息数据合并</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>FDR <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/stats/p.adjust.html" class="external-link">p.adjust</a></span><span class="op">(</span><span class="va">p_value</span>, method <span class="op">=</span> <span class="st">"fdr"</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span>  <span class="co"># 计算FDR值</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">GENE</span>, <span class="fu">everything</span><span class="op">(</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span>  <span class="co"># 对数据框的列进行重新排序</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html" class="external-link">arrange</a></span><span class="op">(</span><span class="va">FDR</span><span class="op">)</span>  <span class="co"># 按FDR值排序</span></span>
<span></span>
<span><span class="co"># 将UTMOST分析结果保存为CSV文件</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/write.table.html" class="external-link">write.csv</a></span><span class="op">(</span><span class="va">dat_utmost_GBJ</span>, file <span class="op">=</span> <span class="st">"./3.UTMOST_GBJ/GBJ_res.csv"</span>, row.names <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre>
<p><img src="images%2Fimg1%2F5.GBJ_RES.jpg" width="80%" style="display: block; margin: auto;"></p>
</div>
<div class="section level4">
<h4 id="进行fusion分析">
<strong>4.进行FUSION分析</strong><a class="anchor" aria-label="anchor" href="#%E8%BF%9B%E8%A1%8Cfusion%E5%88%86%E6%9E%90"></a>
</h4>
<ul>
<li>4.1 数据转换：将GWAS数据转换为FUSION分析.sumstats.gz格式</li>
</ul>
<pre><code><span><span class="fu"><a href="../reference/TWAS_fusion_format_data.html">TWAS_fusion_format_data</a></span><span class="op">(</span></span>
<span>  GWASfile <span class="op">=</span> <span class="st">"./1.data/Migraine_TwosampleMR_FinnGen.txt"</span>,  <span class="co"># 输入的GWAS文件路径</span></span>
<span>  N <span class="op">=</span> <span class="cn">NULL</span>,  <span class="co"># 样本量，如果为空则自动从数据中获取</span></span>
<span>  save_name <span class="op">=</span> <span class="st">"Migraine"</span>,  <span class="co"># 保存的名称</span></span>
<span>  save_path <span class="op">=</span> <span class="st">"./1.data"</span>  <span class="co"># 保存路径</span></span>
<span><span class="op">)</span></span></code></pre>
<p><img src="images%2Fimg1%2F6.sumstats.jpg" width="80%" style="display: block; margin: auto;"></p>
<ul>
<li>4.2 单组织分析：进行单组织的FUSION分析</li>
</ul>
<pre><code><span><span class="co">#此步骤耗时较长，约10余小时...</span></span>
<span><span class="va">TWAS_fusion_Multi</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/TWAS_fusion_Multi_test.html">TWAS_fusion_Multi_test</a></span><span class="op">(</span></span>
<span>  Sumstatsfile <span class="op">=</span> <span class="st">"./1.data/Migraine.sumstats.gz"</span>,  <span class="co"># 输入的结局sumstats.gz格式文件路径</span></span>
<span>  weights_type <span class="op">=</span> <span class="st">"nofilter.pos"</span>,  <span class="co"># 权重数据类型</span></span>
<span>  weights_dir <span class="op">=</span> <span class="st">"./4.FUSION/WEIGHTS/"</span>,  <span class="co"># 权重文件目录路径</span></span>
<span>  resource <span class="op">=</span> <span class="st">"GTEx_v8"</span>,  <span class="co"># 资源数据库</span></span>
<span>  ref_ld_chr <span class="op">=</span> <span class="st">"./4.FUSION/LDREF/1000G.EUR."</span>,  <span class="co"># LD参考文件路径</span></span>
<span>  start_chr <span class="op">=</span> <span class="fl">1</span>,  <span class="co"># 起始染色体</span></span>
<span>  end_chr <span class="op">=</span> <span class="fl">22</span>,  <span class="co"># 结束染色体</span></span>
<span>  coloc_pval <span class="op">=</span> <span class="cn">NULL</span>,  <span class="co"># 共定位p值（如果为空，则不使用共定位分析）</span></span>
<span>  GWASN <span class="op">=</span> <span class="cn">NULL</span>,  <span class="co"># GWAS样本量（如果为空，则自动从数据中获取）</span></span>
<span>  perm <span class="op">=</span> <span class="fl">10000</span>,  <span class="co"># 置换检验迭代次数</span></span>
<span>  PANELN <span class="op">=</span> <span class="cn">NA</span>,  <span class="co"># 面板样本量（如果为空，则自动从数据中获取）</span></span>
<span>  opt_arguments <span class="op">=</span> <span class="cn">NULL</span>,  <span class="co"># 额外参数</span></span>
<span>  remove_MHC <span class="op">=</span> <span class="cn">TRUE</span>,  <span class="co"># 是否去除MHC区域</span></span>
<span>  FDR_method <span class="op">=</span> <span class="st">"fdr"</span>,  <span class="co"># FDR矫正方法</span></span>
<span>  save_name <span class="op">=</span> <span class="st">"Migraine"</span>,  <span class="co"># 保存的名称</span></span>
<span>  save_path <span class="op">=</span> <span class="st">"./4.FUSION"</span>,  <span class="co"># 保存路径</span></span>
<span>  cores <span class="op">=</span> <span class="fl">10</span>  <span class="co"># 运行使用的电脑线程数</span></span>
<span><span class="op">)</span></span></code></pre>
<p><img src="images%2Fimg1%2F7.fusion_res.jpg" width="80%" style="display: block; margin: auto;"></p>
<ul>
<li>4.3 条件联合分析：进行条件联合分析</li>
</ul>
<pre><code><span><span class="va">conditonal_test</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/TWAS_fusion_conditonal_test.html">TWAS_fusion_conditonal_test</a></span><span class="op">(</span></span>
<span>  test_help <span class="op">=</span> <span class="cn">FALSE</span>,  <span class="co"># 是否显示帮助文档信息</span></span>
<span>  Sumstatsfile <span class="op">=</span> <span class="st">"./1.data/Migraine.sumstats.gz"</span>,  <span class="co"># 输入的结局sumstats.gz格式文件路径</span></span>
<span>  inputfile <span class="op">=</span> <span class="st">"./4.FUSION/GTExv8.EUR.Cells_Cultured_fibroblasts.nofilter.pos.sig.dat"</span>,  <span class="co"># 显著性文件路径</span></span>
<span>  ref_ld_chr <span class="op">=</span> <span class="st">"./4.FUSION/LDREF/1000G.EUR."</span>,  <span class="co"># LD参考文件路径</span></span>
<span>  start_chr <span class="op">=</span> <span class="fl">1</span>,  <span class="co"># 起始染色体</span></span>
<span>  end_chr <span class="op">=</span> <span class="fl">22</span>,  <span class="co"># 结束染色体</span></span>
<span>  locus_win_kb <span class="op">=</span> <span class="fl">1000</span>,  <span class="co"># 基因组区域窗口大小（以kb为单位）</span></span>
<span>  build <span class="op">=</span> <span class="fl">38</span>,  <span class="co"># 基因组版本（38代表GRCh38）</span></span>
<span>  plot_legend <span class="op">=</span> <span class="st">"joint"</span>,  <span class="co"># 图例类型</span></span>
<span>  opt_arguments <span class="op">=</span> <span class="cn">NULL</span>,  <span class="co"># 额外参数</span></span>
<span>  save_name <span class="op">=</span> <span class="st">"Migraine"</span>,  <span class="co"># 保存的名称</span></span>
<span>  save_path <span class="op">=</span> <span class="st">"./5.conditonal"</span>,  <span class="co"># 保存路径</span></span>
<span>  cores <span class="op">=</span> <span class="fl">8</span>  <span class="co"># 运行使用的电脑线程数</span></span>
<span><span class="op">)</span></span></code></pre>
<p><img src="images%2Fimg1%2F8.conditonal.png" width="80%" style="display: block; margin: auto;"></p>
</div>
<div class="section level4">
<h4 id="进行magma分析">
<strong>5.进行MAGMA分析</strong><a class="anchor" aria-label="anchor" href="#%E8%BF%9B%E8%A1%8Cmagma%E5%88%86%E6%9E%90"></a>
</h4>
<ul>
<li>5.1 使用MAGMA软件进行基因层面和基因集合层面的关联分析</li>
</ul>
<pre><code><span><span class="va">gene_based_dat</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/MAGMA_gene_based.html">MAGMA_gene_based</a></span><span class="op">(</span></span>
<span>  GWAS_file <span class="op">=</span> <span class="st">"./1.data/Migraine_METAL_FinnGen.txt"</span>,  <span class="co"># 输入的GWAS文件路径</span></span>
<span>  bfile_1000G <span class="op">=</span> <span class="st">"./6.MAGMA/g1000_eur/g1000_eur"</span>,  <span class="co"># 1000G参考数据路径</span></span>
<span>  gene_loc <span class="op">=</span> <span class="st">"./6.MAGMA/ENSGv110.coding.genes.txt"</span>,  <span class="co"># 基因位置文件路径</span></span>
<span>  set_annot <span class="op">=</span> <span class="st">"./6.MAGMA/MSigDB_20231Hs_MAGMA.txt"</span>,  <span class="co"># 基因集合注释文件路径</span></span>
<span>  SNP_P_col <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">3</span>, <span class="fl">10</span><span class="op">)</span>,  <span class="co"># SNP p值列的位置</span></span>
<span>  samplesize_col <span class="op">=</span> <span class="st">"N"</span>,  <span class="co"># 样本量列的名称</span></span>
<span>  save_name <span class="op">=</span> <span class="st">"Migraine"</span>,  <span class="co"># 保存的名称</span></span>
<span>  save_path <span class="op">=</span> <span class="st">"./6.MAGMA"</span>  <span class="co"># 保存路径</span></span>
<span><span class="op">)</span></span></code></pre>
<ul>
<li>5.2 使用MAGMA进行组织特异性分析</li>
</ul>
<pre><code><span><span class="va">Tissue_specific_dat</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/MAGMA_Tissue_specific.html">MAGMA_Tissue_specific</a></span><span class="op">(</span></span>
<span>  genes_raw <span class="op">=</span> <span class="st">"./6.MAGMA/Migraine.genes.raw"</span>,  <span class="co"># 原始基因数据路径</span></span>
<span>  gene_covar <span class="op">=</span> <span class="st">"./6.MAGMA/gtex_v8_ts_avg_log2TPM.txt"</span>,  <span class="co"># 组织特异性表达数据路径</span></span>
<span>  save_name <span class="op">=</span> <span class="st">"Migraine"</span>,  <span class="co"># 保存的名称</span></span>
<span>  save_path <span class="op">=</span> <span class="st">"./6.MAGMA/"</span>  <span class="co"># 保存路径</span></span>
<span><span class="op">)</span></span></code></pre>
<ul>
<li>5.3 对MAGMA基因层面分析结果进行基因注释及曼哈顿图绘制</li>
</ul>
<pre><code><span><span class="va">MAGMA_genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/MAGMA_genes_Manhattanplot.html">MAGMA_genes_Manhattanplot</a></span><span class="op">(</span></span>
<span>  genes_out <span class="op">=</span> <span class="st">"./6.MAGMA/Migraine.genes.out.txt"</span>,  <span class="co"># 基因分析结果文件路径</span></span>
<span>  gene_loc <span class="op">=</span> <span class="st">"./6.MAGMA/ENSGv110.coding.genes.txt"</span>,  <span class="co"># 基因位置文件路径</span></span>
<span>  Manhtn_plot <span class="op">=</span> <span class="cn">TRUE</span>,  <span class="co"># 是否绘制曼哈顿图</span></span>
<span>  threshold_sig <span class="op">=</span> <span class="st">"bonferroni"</span>,  <span class="co"># 显著性阈值调整方法</span></span>
<span>  Manhtn_gene_sig <span class="op">=</span> <span class="fl">10</span>,  <span class="co"># 曼哈顿图的基因标记标签的数目</span></span>
<span>  signal_cex <span class="op">=</span> <span class="fl">1</span>,  <span class="co"># 显著性基因图形圆点的大小</span></span>
<span>  width <span class="op">=</span> <span class="fl">9</span>,  <span class="co"># 图形宽度</span></span>
<span>  height <span class="op">=</span> <span class="fl">7</span>,  <span class="co"># 图形高度</span></span>
<span>  save_name <span class="op">=</span> <span class="st">"MAGMA_gsa"</span>  <span class="co"># 保存的名称</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># 筛选FDR值小于0.05的基因</span></span>
<span><span class="va">MAGMA_genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/subset.html" class="external-link">subset</a></span><span class="op">(</span><span class="va">MAGMA_genes</span>, <span class="va">MAGMA_genes</span><span class="op">$</span><span class="va">P.fdr</span> <span class="op">&lt;</span> <span class="fl">0.05</span><span class="op">)</span></span></code></pre>
<p><img src="images%2Fimg1%2F9.utmost.jpg" width="80%" style="display: block; margin: auto;"></p>
<ul>
<li>5.4 图形可视化</li>
</ul>
<pre><code><span><span class="co"># 对MAGMA通路富集分析结果图形可视化</span></span>
<span><span class="va">plot_dat</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/MAGMA_gsa_barplot.html">MAGMA_gsa_barplot</a></span><span class="op">(</span></span>
<span>  gsa_out <span class="op">=</span> <span class="st">"./6.MAGMA/Migraine.gsa.out.txt"</span>,  <span class="co"># GSA结果文件路径</span></span>
<span>  showNum <span class="op">=</span> <span class="fl">50</span>,  <span class="co"># 显示的前50个通路</span></span>
<span>  X_text_size <span class="op">=</span> <span class="fl">5</span>,  <span class="co"># X轴文本大小</span></span>
<span>  X_text_angle <span class="op">=</span> <span class="fl">80</span>,  <span class="co"># X轴文本角度</span></span>
<span>  Y_text_size <span class="op">=</span> <span class="fl">10</span>,  <span class="co"># Y轴文本大小</span></span>
<span>  save_plot <span class="op">=</span> <span class="cn">TRUE</span>,  <span class="co"># 是否保存图形</span></span>
<span>  pdf_name <span class="op">=</span> <span class="st">"gsa_plot"</span>,  <span class="co"># PDF文件名</span></span>
<span>  width <span class="op">=</span> <span class="fl">9</span>,  <span class="co"># 图形宽度</span></span>
<span>  height <span class="op">=</span> <span class="fl">7</span>,  <span class="co"># 图形高度</span></span>
<span>  save_path <span class="op">=</span> <span class="st">"./6.MAGMA/"</span>  <span class="co"># 保存路径</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># 对MAGMA组织特异性分析结果图形可视化</span></span>
<span><span class="va">Tissue_specific_dat</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/MAGMA_gsa_barplot.html">MAGMA_gsa_barplot</a></span><span class="op">(</span></span>
<span>  gsa_out <span class="op">=</span> <span class="st">"./6.MAGMA/Migraine.Tissue_specific.gsa.out.txt"</span>,  <span class="co"># 组织特异性GSA结果文件路径</span></span>
<span>  showNum <span class="op">=</span> <span class="fl">54</span>,  <span class="co"># 显示的54个组织柱状图</span></span>
<span>  X_text_size <span class="op">=</span> <span class="fl">7</span>,  <span class="co"># X轴文本大小</span></span>
<span>  X_text_angle <span class="op">=</span> <span class="fl">70</span>,  <span class="co"># X轴文本角度</span></span>
<span>  Y_text_size <span class="op">=</span> <span class="fl">10</span>,  <span class="co"># Y轴文本大小</span></span>
<span>  save_plot <span class="op">=</span> <span class="cn">TRUE</span>,  <span class="co"># 是否保存图形</span></span>
<span>  pdf_name <span class="op">=</span> <span class="st">"Tissue_specific_gsa_plot"</span>,  <span class="co"># PDF文件名</span></span>
<span>  width <span class="op">=</span> <span class="fl">9</span>,  <span class="co"># 图形宽度</span></span>
<span>  height <span class="op">=</span> <span class="fl">7</span>,  <span class="co"># 图形高度</span></span>
<span>  save_path <span class="op">=</span> <span class="st">"./6.MAGMA/"</span>  <span class="co"># 保存路径</span></span>
<span><span class="op">)</span></span></code></pre>
<p><img src="images%2Fimg1%2F10.gsa_plot.png" width="80%" style="display: block; margin: auto;"></p>
</div>
<div class="section level4">
<h4 id="绘制韦恩图">
<strong>6.绘制韦恩图</strong><a class="anchor" aria-label="anchor" href="#%E7%BB%98%E5%88%B6%E9%9F%A6%E6%81%A9%E5%9B%BE"></a>
</h4>
<ul>
<li>对UTMOST、FUSION和MAGMA分析结果进行交集分析</li>
</ul>
<pre><code><span><span class="va">venn_plot</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/venn_plot.html">venn_plot</a></span><span class="op">(</span></span>
<span>  files <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"./7.venn/UTMOST.txt"</span>, <span class="st">"./7.venn/FUSION.txt"</span>, <span class="st">"./7.venn/MAGMA.txt"</span><span class="op">)</span>,  <span class="co"># 输入的结果文件路径</span></span>
<span>  save_plot <span class="op">=</span> <span class="cn">TRUE</span>,  <span class="co"># 是否保存图形</span></span>
<span>  save_name <span class="op">=</span> <span class="st">"venn_plot"</span>,  <span class="co"># 保存的名称</span></span>
<span>  width <span class="op">=</span> <span class="fl">9</span>,  <span class="co"># 图形宽度</span></span>
<span>  height <span class="op">=</span> <span class="fl">7</span>,  <span class="co"># 图形高度</span></span>
<span>  save_path <span class="op">=</span> <span class="st">"./7.venn"</span>  <span class="co"># 保存路径</span></span>
<span><span class="op">)</span></span></code></pre>
<p><img src="images%2Fimg1%2F11.venn.jpg" width="80%" style="display: block; margin: auto;"></p>
</div>
<div class="section level4">
<h4 id="孟德尔随机化mr分析">
<strong>7.孟德尔随机化（MR）分析</strong><a class="anchor" aria-label="anchor" href="#%E5%AD%9F%E5%BE%B7%E5%B0%94%E9%9A%8F%E6%9C%BA%E5%8C%96mr%E5%88%86%E6%9E%90"></a>
</h4>
<ul>
<li>7.1 获取GTEx V8组织、样本量信息</li>
</ul>
<pre><code><span><span class="va">GTEx_V8_samplesize</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/data_info_GTEx_V8_samplesize.html">data_info_GTEx_V8_samplesize</a></span><span class="op">(</span><span class="op">)</span></span></code></pre>
<ul>
<li>7.2 获取不同组织的cis-eQTL数据</li>
</ul>
<pre><code><span><span class="va">ciseQTL1</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_ciseQTL_GTEx_V8_Online.html">get_ciseQTL_GTEx_V8_Online</a></span><span class="op">(</span></span>
<span>  Gene_name <span class="op">=</span> <span class="st">"REV1"</span>,  <span class="co"># 基因名称</span></span>
<span>  tissues <span class="op">=</span> <span class="st">"Whole Blood"</span>,  <span class="co"># 组织类型</span></span>
<span>  cis_wind_kb <span class="op">=</span> <span class="fl">1000</span>,  <span class="co"># cis区域窗口大小（以kb为单位）</span></span>
<span>  build <span class="op">=</span> <span class="fl">38</span>,  <span class="co"># 基因组版本（38代表GRCh38）</span></span>
<span>  save_path <span class="op">=</span> <span class="st">"./8.MR"</span>  <span class="co"># 保存路径</span></span>
<span><span class="op">)</span></span>
<span><span class="va">ciseQTL1</span><span class="op">$</span><span class="va">gene</span> <span class="op">&lt;-</span> <span class="st">"REV1(Whole Blood)"</span>  <span class="co"># 添加组织信息</span></span>
<span></span>
<span><span class="va">ciseQTL2</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_ciseQTL_GTEx_V8_Online.html">get_ciseQTL_GTEx_V8_Online</a></span><span class="op">(</span></span>
<span>  Gene_name <span class="op">=</span> <span class="st">"REV1"</span>,  <span class="co"># 基因名称</span></span>
<span>  tissues <span class="op">=</span> <span class="st">"Cells - Cultured fibroblasts"</span>,  <span class="co"># 组织类型</span></span>
<span>  cis_wind_kb <span class="op">=</span> <span class="fl">1000</span>,  <span class="co"># cis区域窗口大小（以kb为单位）</span></span>
<span>  build <span class="op">=</span> <span class="fl">38</span>,  <span class="co"># 基因组版本（38代表GRCh38）</span></span>
<span>  save_path <span class="op">=</span> <span class="st">"./8.MR"</span>  <span class="co"># 保存路径</span></span>
<span><span class="op">)</span></span>
<span><span class="va">ciseQTL2</span><span class="op">$</span><span class="va">gene</span> <span class="op">&lt;-</span> <span class="st">"REV1(Cells - Cultured fibroblasts)"</span>  <span class="co"># 添加组织信息</span></span>
<span></span>
<span><span class="va">ciseQTL3</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/get_ciseQTL_GTEx_V8_Online.html">get_ciseQTL_GTEx_V8_Online</a></span><span class="op">(</span></span>
<span>  Gene_name <span class="op">=</span> <span class="st">"SREBF2"</span>,  <span class="co"># 基因名称</span></span>
<span>  tissues <span class="op">=</span> <span class="st">"Testis"</span>,  <span class="co"># 组织类型</span></span>
<span>  cis_wind_kb <span class="op">=</span> <span class="fl">1000</span>,  <span class="co"># cis区域窗口大小（以kb为单位）</span></span>
<span>  build <span class="op">=</span> <span class="fl">38</span>,  <span class="co"># 基因组版本（38代表GRCh38）</span></span>
<span>  save_path <span class="op">=</span> <span class="st">"./8.MR"</span>  <span class="co"># 保存路径</span></span>
<span><span class="op">)</span></span>
<span><span class="va">ciseQTL3</span><span class="op">$</span><span class="va">gene</span> <span class="op">&lt;-</span> <span class="st">"SREBF2(Testis)"</span>  <span class="co"># 添加组织信息</span></span>
<span></span>
<span><span class="co"># 合并所有的cis-eQTL数据</span></span>
<span><span class="va">ciseQTL</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">rbind</a></span><span class="op">(</span><span class="va">ciseQTL1</span>, <span class="va">ciseQTL2</span>, <span class="va">ciseQTL3</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># 按基因分割数据成list</span></span>
<span><span class="va">ciseQTL</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/split.html" class="external-link">split</a></span><span class="op">(</span><span class="va">ciseQTL</span>, <span class="va">ciseQTL</span><span class="op">$</span><span class="va">gene</span><span class="op">)</span>  </span>
<span></span>
<span><span class="co"># 保存合并后的cis-eQTL数据</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/save.html" class="external-link">save</a></span><span class="op">(</span><span class="va">ciseQTL</span>, file <span class="op">=</span> <span class="st">"./8.MR/ciseQTL.Rdata"</span><span class="op">)</span></span></code></pre>
<ul>
<li>7.3 格式化数据并进行去除连锁不平衡</li>
</ul>
<pre><code><span><span class="va">exp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="kw">for</span> <span class="op">(</span><span class="va">i</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">ciseQTL</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="va">dat</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/format_dat.html">format_dat</a></span><span class="op">(</span></span>
<span>    <span class="va">ciseQTL</span><span class="op">[[</span><span class="va">i</span><span class="op">]</span><span class="op">]</span>,  <span class="co"># 输入的cis-eQTL数据</span></span>
<span>    type <span class="op">=</span> <span class="st">"exposure"</span>,  <span class="co"># 数据类型为暴露数据</span></span>
<span>    phenotype_col <span class="op">=</span> <span class="st">"gene"</span>,  <span class="co"># 基因列的名称</span></span>
<span>    snp_col <span class="op">=</span> <span class="st">"SNP"</span>,  <span class="co"># SNP列的名称</span></span>
<span>    beta_col <span class="op">=</span> <span class="st">"beta"</span>,  <span class="co"># beta系数列的名称</span></span>
<span>    se_col <span class="op">=</span> <span class="st">"se"</span>,  <span class="co"># 标准误列的名称</span></span>
<span>    eaf_col <span class="op">=</span> <span class="st">"eaf"</span>,  <span class="co"># 等位基因频率列的名称</span></span>
<span>    effect_allele_col <span class="op">=</span> <span class="st">"effect_allele"</span>,  <span class="co"># 效应等位基因列的名称</span></span>
<span>    other_allele_col <span class="op">=</span> <span class="st">"other_allele"</span>,  <span class="co"># 非效应等位基因列的名称</span></span>
<span>    pval_col <span class="op">=</span> <span class="st">"pval"</span>,  <span class="co"># p值列的名称</span></span>
<span>    samplesize_col <span class="op">=</span> <span class="st">"N"</span>,  <span class="co"># 样本量列的名称</span></span>
<span>    chr_col <span class="op">=</span> <span class="st">"CHR"</span>,  <span class="co"># 染色体列的名称</span></span>
<span>    pos_col <span class="op">=</span> <span class="st">"BP"</span>  <span class="co"># 位点列的名称</span></span>
<span>  <span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>    <span class="fu"><a href="../reference/clump_data_local_Online.html">clump_data_local_Online</a></span><span class="op">(</span></span>
<span>      snp_col <span class="op">=</span> <span class="st">"SNP"</span>,  <span class="co"># SNP列的名称</span></span>
<span>      pval_col <span class="op">=</span> <span class="st">"pval.exposure"</span>,  <span class="co"># p值列的名称</span></span>
<span>      clump_method <span class="op">=</span> <span class="st">"PVAL"</span>,  <span class="co"># clump方法</span></span>
<span>      beta_col <span class="op">=</span> <span class="st">"beta.exposure"</span>,  <span class="co"># beta系数列的名称</span></span>
<span>      se_col <span class="op">=</span> <span class="st">"se.exposure"</span>,  <span class="co"># 标准误列的名称</span></span>
<span>      clump_pval <span class="op">=</span> <span class="fl">5e-08</span>,  <span class="co"># clump p值阈值</span></span>
<span>      clump_kb <span class="op">=</span> <span class="fl">10000</span>,  <span class="co"># clump窗口大小（以kb为单位）</span></span>
<span>      clump_r2 <span class="op">=</span> <span class="fl">0.001</span>,  <span class="co"># clump R^2阈值</span></span>
<span>      pop <span class="op">=</span> <span class="st">"EUR"</span>,  <span class="co"># 参考人群为欧洲人群</span></span>
<span>      bfile_1000G <span class="op">=</span> <span class="st">"./1.data/1000G/EUR"</span>  <span class="co"># 1000G参考数据路径</span></span>
<span>    <span class="op">)</span></span>
<span>  <span class="va">exp</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">rbind</a></span><span class="op">(</span><span class="va">exp</span>, <span class="va">dat</span><span class="op">)</span>  <span class="co"># 合并所有格式化和clump后的数据</span></span>
<span><span class="op">}</span></span></code></pre>
<ul>
<li>7.4 本地GWAS数据获取代理SNP</li>
</ul>
<pre><code><span><span class="va">out</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/modified_proxy_1000G.html">modified_proxy_1000G</a></span><span class="op">(</span></span>
<span>  snps <span class="op">=</span> <span class="va">exp</span><span class="op">$</span><span class="va">SNP</span>,  <span class="co"># SNP列表</span></span>
<span>  GWASfile <span class="op">=</span> <span class="st">"./1.data/Migraine_TwosampleMR_FinnGen.txt"</span>,  <span class="co"># GWAS文件路径</span></span>
<span>  proxies <span class="op">=</span> <span class="cn">TRUE</span>,  <span class="co"># 是否寻找代理SNP</span></span>
<span>  rsq <span class="op">=</span> <span class="fl">0.8</span>,  <span class="co"># 代理SNP的R^2阈值</span></span>
<span>  kb <span class="op">=</span> <span class="fl">5000</span>,  <span class="co"># 代理SNP的窗口大小（以kb为单位）</span></span>
<span>  nsnp <span class="op">=</span> <span class="fl">5000</span>,  <span class="co"># 代理SNP的最大数量</span></span>
<span>  maf_threshold <span class="op">=</span> <span class="fl">0.3</span>,  <span class="co"># 最小等位基因频率阈值</span></span>
<span>  bfile_1000G <span class="op">=</span> <span class="st">"./1.data/1000G/EUR"</span>  <span class="co"># 1000G参考数据路径</span></span>
<span><span class="op">)</span></span></code></pre>
<ul>
<li>7.5 对暴露和结果数据进行harmonisation（协调化）</li>
</ul>
<pre><code><span><span class="va">dat</span> <span class="op">&lt;-</span> <span class="fu">TwoSampleMR</span><span class="fu">::</span><span class="fu"><a href="https://mrcieu.github.io/TwoSampleMR/reference/harmonise_data.html" class="external-link">harmonise_data</a></span><span class="op">(</span><span class="va">exp</span>, <span class="va">out</span><span class="op">)</span></span></code></pre>
<ul>
<li>7.6 MR分析与OR（比值比）转换</li>
</ul>
<pre><code><span><span class="co"># 进行孟德尔随机化（MR）分析</span></span>
<span><span class="va">res</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/xQTL_mr.html">xQTL_mr</a></span><span class="op">(</span><span class="va">dat</span>, FDR_method <span class="op">=</span> <span class="cn">NULL</span>, PVE <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># 生成OR（比值比）结果</span></span>
<span><span class="va">or</span> <span class="op">&lt;-</span> <span class="fu">TwoSampleMR</span><span class="fu">::</span><span class="fu"><a href="https://mrcieu.github.io/TwoSampleMR/reference/generate_odds_ratios.html" class="external-link">generate_odds_ratios</a></span><span class="op">(</span><span class="va">res</span><span class="op">)</span></span></code></pre>
<p><img src="images%2Fimg1%2F12.MR.jpg" width="100%" style="display: block; margin: auto;"></p>
<ul>
<li>7.7 绘制森林图</li>
</ul>
<pre><code><span><span class="fu"><a href="../reference/xQTL_forest.html">xQTL_forest</a></span><span class="op">(</span></span>
<span>  <span class="va">res</span>,  <span class="co"># MR分析结果</span></span>
<span>  pvalsig <span class="op">=</span> <span class="cn">NULL</span>,  <span class="co"># 显著性p值阈值</span></span>
<span>  ci_col <span class="op">=</span> <span class="st">"#000000"</span>,  <span class="co"># 置信区间颜色</span></span>
<span>  ci_fill <span class="op">=</span> <span class="st">"#000000"</span>,  <span class="co"># 置信区间填充色</span></span>
<span>  ci_lwd <span class="op">=</span> <span class="fl">1</span>,  <span class="co"># 置信区间线宽</span></span>
<span>  xlim <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.6</span>, <span class="fl">1.2</span><span class="op">)</span>,  <span class="co"># x轴范围</span></span>
<span>  ticks_at <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.6</span>, <span class="fl">0.7</span>, <span class="fl">0.8</span>, <span class="fl">0.9</span>, <span class="fl">1</span>, <span class="fl">1.1</span>, <span class="fl">1.2</span><span class="op">)</span>,  <span class="co"># x轴刻度</span></span>
<span>  save_plot <span class="op">=</span> <span class="cn">TRUE</span>,  <span class="co"># 是否保存图形</span></span>
<span>  plot_pdf <span class="op">=</span> <span class="st">"forest_plot"</span>,  <span class="co"># PDF文件名</span></span>
<span>  width <span class="op">=</span> <span class="fl">8</span>,  <span class="co"># 图形宽度</span></span>
<span>  height <span class="op">=</span> <span class="fl">5</span>,  <span class="co"># 图形高度</span></span>
<span>  save_path <span class="op">=</span> <span class="st">"./8.MR"</span>  <span class="co"># 保存路径</span></span>
<span><span class="op">)</span></span></code></pre>
<p><img src="images%2Fimg1%2F13.forest.png" width="100%" style="display: block; margin: auto;"></p>
</div>
<div class="section level4">
<h4 id="共定位分析">
<strong>8.共定位分析</strong><a class="anchor" aria-label="anchor" href="#%E5%85%B1%E5%AE%9A%E4%BD%8D%E5%88%86%E6%9E%90"></a>
</h4>
<ul>
<li>8.1 读取结局数据GWAS结果数据</li>
</ul>
<pre><code><span><span class="va">outcome_dat</span> <span class="op">&lt;-</span> <span class="fu">vroom</span><span class="fu">::</span><span class="fu"><a href="https://vroom.r-lib.org/reference/vroom.html" class="external-link">vroom</a></span><span class="op">(</span><span class="st">"./1.data/Migraine_TwosampleMR_FinnGen.txt"</span>, show_col_types <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre>
<ul>
<li>8.2 加载之前保存的cis-eQTL数据</li>
</ul>
<pre><code><span><span class="co"># 加载之前保存的cis-eQTL数据</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="st">"./8.MR/ciseQTL.Rdata"</span><span class="op">)</span></span></code></pre>
<ul>
<li>8.3 循环进行共定位分析</li>
</ul>
<pre><code><span><span class="va">coloc_summary</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># 遍历每个基因进行共定位分析</span></span>
<span><span class="kw">for</span> <span class="op">(</span><span class="va">j</span> <span class="kw">in</span> <span class="fu"><a href="https://rdrr.io/r/base/names.html" class="external-link">names</a></span><span class="op">(</span><span class="va">ciseQTL</span><span class="op">)</span><span class="op">)</span> <span class="op">{</span></span>
<span>  <span class="va">dat_coloc</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/coloc_abf.html">coloc_abf</a></span><span class="op">(</span></span>
<span>    exposure_dat <span class="op">=</span> <span class="va">ciseQTL</span><span class="op">[[</span><span class="va">j</span><span class="op">]</span><span class="op">]</span>,  <span class="co"># 暴露数据</span></span>
<span>    outcome_dat <span class="op">=</span> <span class="va">outcome_dat</span>,  <span class="co"># 结局数据</span></span>
<span>    type_exposure <span class="op">=</span> <span class="st">"quant"</span>,  <span class="co"># 暴露类型为定量</span></span>
<span>    col_pvalues_exposure <span class="op">=</span> <span class="st">"pval"</span>,  <span class="co"># 暴露数据中的p值列</span></span>
<span>    col_N_exposure <span class="op">=</span> <span class="st">"N"</span>,  <span class="co"># 暴露数据中的样本量列</span></span>
<span>    col_MAF_exposure <span class="op">=</span> <span class="st">"MAF"</span>,  <span class="co"># 暴露数据中的等位基因频率列</span></span>
<span>    col_beta_exposure <span class="op">=</span> <span class="st">"beta"</span>,  <span class="co"># 暴露数据中的beta系数列</span></span>
<span>    col_se_exposure <span class="op">=</span> <span class="st">"se"</span>,  <span class="co"># 暴露数据中的标准误列</span></span>
<span>    col_snp_exposure <span class="op">=</span> <span class="st">"SNP"</span>,  <span class="co"># 暴露数据中的SNP列</span></span>
<span>    col_chr_exposure <span class="op">=</span> <span class="st">"CHR"</span>,  <span class="co"># 暴露数据中的染色体列</span></span>
<span>    col_pos_exposure <span class="op">=</span> <span class="st">"BP"</span>,  <span class="co"># 暴露数据中的位点列</span></span>
<span>    prevalence_exposure <span class="op">=</span> <span class="cn">NA</span>,  <span class="co"># 暴露数据中的患病率</span></span>
<span>    type_outcome <span class="op">=</span> <span class="st">"cc"</span>,  <span class="co"># 结局类型为病例对照</span></span>
<span>    col_pvalues_outcome <span class="op">=</span> <span class="st">"pval.outcome"</span>,  <span class="co"># 结局数据中的p值列</span></span>
<span>    col_N_outcome <span class="op">=</span> <span class="st">"samplesize.outcome"</span>,  <span class="co"># 结局数据中的样本量列</span></span>
<span>    col_MAF_outcome <span class="op">=</span> <span class="cn">NA</span>,  <span class="co"># 结局数据中的等位基因频率列</span></span>
<span>    col_beta_outcome <span class="op">=</span> <span class="st">"beta.outcome"</span>,  <span class="co"># 结局数据中的beta系数列</span></span>
<span>    col_se_outcome <span class="op">=</span> <span class="st">"se.outcome"</span>,  <span class="co"># 结局数据中的标准误列</span></span>
<span>    col_snp_outcome <span class="op">=</span> <span class="st">"SNP"</span>,  <span class="co"># 结局数据中的SNP列</span></span>
<span>    prevalence_outcome <span class="op">=</span> <span class="cn">NA</span>,  <span class="co"># 结局数据中的患病率</span></span>
<span>    save_stacked_dat <span class="op">=</span> <span class="cn">TRUE</span>,  <span class="co"># 是否保存绘制连锁区域图数据</span></span>
<span>    build <span class="op">=</span> <span class="fl">38</span>,  <span class="co"># 基因组版本（38代表GRCh38）</span></span>
<span>    save_locus <span class="op">=</span> <span class="cn">FALSE</span>,  <span class="co"># 是否保存locus数据</span></span>
<span>    title1 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"eQTL "</span>, <span class="va">j</span><span class="op">)</span>,  <span class="co"># 图形标题1</span></span>
<span>    title2 <span class="op">=</span> <span class="st">"GWAS Migraine"</span>,  <span class="co"># 图形标题2    请自行按照实际GWAS名字修改</span></span>
<span>    width <span class="op">=</span> <span class="fl">8</span>,  <span class="co"># 图形宽度</span></span>
<span>    height <span class="op">=</span> <span class="fl">5</span>,  <span class="co"># 图形高度</span></span>
<span>    plot_pdf <span class="op">=</span> <span class="st">"locuscompare"</span>,  <span class="co"># PDF文件名</span></span>
<span>    save_path <span class="op">=</span> <span class="st">"./9.coloc"</span>  <span class="co"># 保存路径</span></span>
<span>  <span class="op">)</span></span>
<span></span>
<span>  <span class="co"># 提取和处理共定位分析结果</span></span>
<span>  <span class="va">summary</span> <span class="op">&lt;-</span> <span class="va">dat_coloc</span><span class="op">[[</span><span class="st">"summary"</span><span class="op">]</span><span class="op">]</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>gene <span class="op">=</span> <span class="va">j</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">gene</span>, <span class="fu">everything</span><span class="op">(</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span>  <span class="co"># 绘制和保存共定位连锁区域结果图</span></span>
<span>  <span class="va">dat_assoc</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/coloc_stack_assoc_plot.html">coloc_stack_assoc_plot</a></span><span class="op">(</span></span>
<span>    assoc <span class="op">=</span> <span class="va">assoc</span>,</span>
<span>    bfile_1000G <span class="op">=</span> <span class="st">"./1.data/1000G/EUR"</span>,  <span class="co"># 1000G参考数据路径</span></span>
<span>    SNP <span class="op">=</span> <span class="va">SigSNP</span>,  <span class="co"># 显著SNP</span></span>
<span>    build <span class="op">=</span> <span class="fl">38</span>,  <span class="co"># 基因组版本（38代表GRCh38）</span></span>
<span>    title1 <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="st">"eQTL "</span>, <span class="va">j</span><span class="op">)</span>,  <span class="co"># 图形标题1</span></span>
<span>    title2 <span class="op">=</span> <span class="st">"GWAS Migraine"</span>,  <span class="co"># 图形标题2</span></span>
<span>    save_plot <span class="op">=</span> <span class="cn">TRUE</span>,  <span class="co"># 是否保存图形</span></span>
<span>    width <span class="op">=</span> <span class="fl">9</span>,  <span class="co"># 图形宽度</span></span>
<span>    height <span class="op">=</span> <span class="fl">10</span>,  <span class="co"># 图形高度</span></span>
<span>    plot_pdf <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste0</a></span><span class="op">(</span><span class="va">j</span>, <span class="st">"_plot"</span><span class="op">)</span>,  <span class="co"># PDF文件名</span></span>
<span>    save_path <span class="op">=</span> <span class="st">"./9.coloc"</span>  <span class="co"># 保存路径</span></span>
<span>  <span class="op">)</span></span>
<span></span>
<span>  <span class="co"># 合并所有共定位分析结果</span></span>
<span>  <span class="va">coloc_summary</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">rbind</a></span><span class="op">(</span><span class="va">coloc_summary</span>, <span class="va">summary</span><span class="op">)</span></span>
<span><span class="op">}</span></span></code></pre>
<ul>
<li>8.4 打印和保存共定位分析总结</li>
</ul>
<pre><code><span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">coloc_summary</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/write.table.html" class="external-link">write.csv</a></span><span class="op">(</span><span class="va">coloc_summary</span>, file <span class="op">=</span> <span class="st">"./9.coloc/coloc_res.csv"</span>, row.names <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span></span></code></pre>
<p><img src="images%2Fimg1%2F12.coloc.jpg" width="100%" style="display: block; margin: auto;"></p>
<pre><code><span><span class="co"># 还可以联合SMR、FOCUS、TWAS其他算法、生信分析及湿实验等方法，丰富文章内容！</span></span>
<span></span>
<span><span class="co">## 感谢各位小伙伴使用...</span></span></code></pre>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by YB H.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
